--- sch_sfb.c.orig	2011-05-07 13:04:58.809286032 -0600
+++ sch_sfb.c	2011-05-07 13:05:26.729734059 -0600
@@ -412,6 +412,23 @@
 	return ret;
 }
 
+/* #if LINUX_KERNEL_VERSION < something */
+
+static inline void bstats_update(struct gnet_stats_basic_packed *bstats,
+                                 const struct sk_buff *skb)
+{
+        bstats->bytes += qdisc_pkt_len(skb);
+        bstats->packets += skb_is_gso(skb) ? skb_shinfo(skb)->gso_segs : 1;
+}
+
+
+static inline void qdisc_bstats_update(struct Qdisc *sch,
+                                       const struct sk_buff *skb)
+{
+        bstats_update(&sch->bstats, skb);
+}
+/* #endif */
+
 static struct sk_buff *sfb_dequeue(struct Qdisc *sch)
 {
 	struct sfb_sched_data *q = qdisc_priv(sch);
