--- a/SConstruct
+++ b/SConstruct
@@ -358,20 +358,8 @@ if env['ncurses']:
         ncurseslibs= [ '-lncurses' ]
 
 if env['usb']:
-    # In FreeBSD except version 7, USB libraries are in the base system
-    if config.CheckPKG('libusb-1.0'):
-        confdefs.append("#define HAVE_LIBUSB 1\n")
-        try:
-            usblibs = pkg_config('libusb-1.0')
-        except OSError:
-            print "pkg_config is confused about the state of libusb-1.0."
-            usblibs = []
-    elif sys.platform.startswith("freebsd"):
-        confdefs.append("#define HAVE_LIBUSB 1\n")
-        usblibs = [ "-lusb"]
-    else:
-        confdefs.append("/* #undef HAVE_LIBUSB */\n")
-        usblibs = []
+    confdefs.append("#define HAVE_LIBUSB 1\n")
+    usblibs = [ "-lusb-1.0"]
 else:
     confdefs.append("/* #undef HAVE_LIBUSB */\n")
     usblibs = []
