--- time-1.7/resuse.h.orig	2011-05-06 15:16:20.850301203 -0600
+++ time-1.7/resuse.h	2011-05-06 15:16:31.840485071 -0600
@@ -31,12 +31,11 @@
 #endif
 
 #if HAVE_SYS_RUSAGE_H
-/* This rusage structure measures nanoseconds instead of microseconds.  */
-# define TV_MSEC tv_nsec / 1000000
 # include <sys/rusage.h>
+# define TV_MSEC tv_nsec / 1000000
 #else
 # define TV_MSEC tv_usec / 1000
-# if HAVE_WAIT3
+# if HAVE_WAIT3 | HAVE_SYS_RESOURCE_H
 #  include <sys/resource.h>
 # else
 /* Process resource usage structure.  */
