--- a/configure.in
+++ b/configure.in
@@ -787,7 +787,25 @@ int main() {
 		[AC_MSG_RESULT(yes)
 		have_ecdsa="yes"],
 		[AC_MSG_RESULT(no)
-		have_ecdsa="no"])
+		have_ecdsa="no"],
+		[AC_MSG_RESULT(using --with-ecdsa)])
+		AC_ARG_WITH(ecdsa, , with_ecdsa="$withval", with_ecdsa="auto")
+		case "$with_ecdsa" in
+		yes)
+		case "$have_ecdsa" in
+		    no)  AC_MSG_ERROR([ecdsa not supported]) ;;
+		    *)  have_ecdsa=yes ;;
+		    esac
+		    ;;
+		no)
+		    have_ecdsa=no ;;
+		*)
+		    case "$have_ecdsa" in
+		    yes|no) ;;
+		    *) AC_MSG_ERROR([need --with-ecdsa=[[yes or no]]]) ;;
+		    esac
+		    ;;
+		esac
 		case $have_ecdsa in
 		yes)
 			OPENSSL_ECDSA="yes"
