Index: bind9/lib/bind9/check.c
diff -u bind9/lib/bind9/check.c:1.125 bind9/lib/bind9/check.c:1.125.14.4
--- bind9/lib/bind9/check.c:1.125	Fri Jan  7 23:47:07 2011
+++ bind9/lib/bind9/check.c	Thu May  5 18:02:09 2011
@@ -2078,6 +2089,7 @@
 	cfg_aclconfctx_t actx;
 	const cfg_obj_t *obj;
 	isc_boolean_t enablednssec, enablevalidation;
+	const char *valstr = "no";
 
 	/*
 	 * Check that all zone statements are syntactically correct and
@@ -2203,14 +2215,21 @@
 		(void)cfg_map_get(voptions, "dnssec-validation", &obj);
 	if (obj == NULL)
 		(void)cfg_map_get(config, "dnssec-validation", &obj);
-	if (obj == NULL)
-		enablevalidation = ISC_FALSE;	/* XXXMPA Change for 9.5. */
-	else
+	if (obj == NULL) {
+		enablevalidation = ISC_TRUE;
+		valstr = "yes";
+	} else if (cfg_obj_isboolean(obj)) {
 		enablevalidation = cfg_obj_asboolean(obj);
+		valstr = enablevalidation ? "yes" : "no";
+	} else {
+		enablevalidation = ISC_TRUE;
+		valstr = "auto";
+	}
 
 	if (enablevalidation && !enablednssec)
 		cfg_obj_log(obj, logctx, ISC_LOG_WARNING,
-			    "'dnssec-validation yes;' and 'dnssec-enable no;'");
+			    "'dnssec-validation %s;' and 'dnssec-enable no;'",
+			    valstr);
 
 	/*
 	 * Check trusted-keys and managed-keys.
