--- a/csrc/lib/dir.mk
+++ b/csrc/lib/dir.mk
@@ -73,8 +73,8 @@ shlib: $(SHLIBNAME)
 
 lib: libccn.a
 
-test: default keystore_check encodedecodetest
-	./encodedecodetest -o /dev/null
+test:
+	@echo "Sorry, can't run tests when cross compiling"
 
 dtag_check: _always
 	@./gen_dtag_table 2>/dev/null | diff - ccn_dtag_table.c | grep '^[<]' >/dev/null && echo '*** Warning: ccn_dtag_table.c may be out of sync with tagnames.cvsdict' || :
