--- a/csrc/lib/dir.mk
+++ b/csrc/lib/dir.mk
@@ -56,8 +56,7 @@ all: basicparsetest ccn_verifysig
 
 install: install_headers
 install_headers:
-	test -d $(INSTALL_INCLUDE)
-	mkdir -p $(INSTALL_INCLUDE)/ccn
+	test -d $(INSTALL_INCLUDE) || mkdir -p $(INSTALL_INCLUDE)/ccn
 	for i in `cd ../include/ccn && echo *.h`; do                \
 	    cmp -s ../include/ccn/$$i $(INSTALL_INCLUDE)/ccn/$$i || \
 	        cp ../include/ccn/$$i $(INSTALL_INCLUDE)/ccn/$$i || \
--- a/csrc/subr.mk
+++ b/csrc/subr.mk
@@ -31,11 +31,11 @@ $(OBJDIR)/dir.mk: dir.mk
 	cp -p dir.mk $(OBJDIR)/dir.mk
 
 install_libs: $(LIBS)
-	test -d $(INSTALL_LIB)
+	test -d $(INSTALL_LIB) || mkdir -p $(INSTALL_LIB)
 	for i in $(LIBS) ""; do test -z "$$i" || $(INSTALL) $$i $(INSTALL_LIB); done
 
 install_programs: $(INSTALLED_PROGRAMS)
-	test -d $(INSTALL_BIN)
+	test -d $(INSTALL_BIN) || mkdir -p $(INSTALL_BIN)
 	for i in $(INSTALLED_PROGRAMS) ""; do test -z "$$i" || $(INSTALL) $$i $(INSTALL_BIN); done
 
 install: install_libs install_programs
