From 6df1575a41f105e8b4155aad8b7c94c0ecb64ab0 Mon Sep 17 00:00:00 2001
From: Dave Taht <dave.taht@bufferbloat.net>
Date: Mon, 20 Feb 2012 21:36:58 +0100
Subject: [PATCH] make the install dir always

---
 csrc/lib/dir.mk |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/csrc/lib/dir.mk b/csrc/lib/dir.mk
index 33de73b..5bfad4e 100644
--- a/csrc/lib/dir.mk
+++ b/csrc/lib/dir.mk
@@ -59,7 +59,6 @@ all: basicparsetest ccn_verifysig
 
 install: install_headers
 install_headers:
-	@test -d $(INSTALL_INCLUDE) || (echo $(INSTALL_INCLUDE) does not exist.  Please mkdir -p $(INSTALL_INCLUDE) if this is what you intended. && exit 2)
 	mkdir -p $(INSTALL_INCLUDE)/ccn
 	for i in `cd ../include/ccn && echo *.h`; do                \
 	    cmp -s ../include/ccn/$$i $(INSTALL_INCLUDE)/ccn/$$i || \
-- 
1.7.5.4

