--- a/csrc/lib/dir.mk
+++ b/csrc/lib/dir.mk
@@ -59,7 +59,6 @@ all: ccn_verifysig
 
 install: install_headers
 install_headers:
-	@test -d $(DINST_INC) || (echo $(DINST_INC) does not exist.  Please mkdir -p $(DINST_INC) if this is what you intended. && exit 2)
 	mkdir -p $(DINST_INC)/ccn
 	for i in `cd ../include/ccn && echo *.h`; do                \
 	    cmp -s ../include/ccn/$$i $(DINST_INC)/ccn/$$i || \
--- a/csrc/subr.mk
+++ b/csrc/subr.mk
@@ -31,11 +31,11 @@ $(OBJDIR)/dir.mk: dir.mk
 	cp -p dir.mk $(OBJDIR)/dir.mk
 
 install_libs: $(LIBS)
-	@test -d $(DINST_LIB) || (echo $(DINST_LIB) does not exist.  Please mkdir -p $(DINST_LIB) if this is what you intended. && exit 2)
+	mkdir -p $(DINST_LIB)
 	for i in $(LIBS) ""; do test -z "$$i" || $(INSTALL) $$i $(DINST_LIB); done
 
 install_programs: $(INSTALLED_PROGRAMS)
-	@test -d $(DINST_BIN) || (echo $(DINST_BIN) does not exist.  Please mkdir -p $(DINST_BIN) if this is what you intended. && exit 2)
+	mkdir -p $(DINST_BIN)
 	for i in $(INSTALLED_PROGRAMS) ""; do test -z "$$i" || $(INSTALL) $$i $(DINST_BIN); done
 
 install: install_libs install_programs
