From 6df1575a41f105e8b4155aad8b7c94c0ecb64ab0 Mon Sep 17 00:00:00 2001
From: Dave Taht <dave.taht@bufferbloat.net>
Date: Mon, 20 Feb 2012 21:36:58 +0100
Subject: [PATCH] make the install dir always

---
 csrc/lib/dir.mk |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

--- a/csrc/lib/dir.mk
+++ b/csrc/lib/dir.mk
@@ -59,7 +59,6 @@ all: basicparsetest ccn_verifysig
 
 install: install_headers
 install_headers:
-	@test -d $(INSTALL_INCLUDE) || (echo $(INSTALL_INCLUDE) does not exist.  Please mkdir -p $(INSTALL_INCLUDE) if this is what you intended. && exit 2)
 	mkdir -p $(INSTALL_INCLUDE)/ccn
 	for i in `cd ../include/ccn && echo *.h`; do                \
 	    cmp -s ../include/ccn/$$i $(INSTALL_INCLUDE)/ccn/$$i || \
--- a/csrc/subr.mk
+++ b/csrc/subr.mk
@@ -31,11 +31,11 @@ $(OBJDIR)/dir.mk: dir.mk
 	cp -p dir.mk $(OBJDIR)/dir.mk
 
 install_libs: $(LIBS)
-	@test -d $(INSTALL_LIB) || (echo $(INSTALL_LIB) does not exist.  Please mkdir -p $(INSTALL_LIB) if this is what you intended. && exit 2)
+	mkdir -p $(INSTALL_LIB)
 	for i in $(LIBS) ""; do test -z "$$i" || $(INSTALL) $$i $(INSTALL_LIB); done
 
 install_programs: $(INSTALLED_PROGRAMS)
-	@test -d $(INSTALL_BIN) || (echo $(INSTALL_BIN) does not exist.  Please mkdir -p $(INSTALL_BIN) if this is what you intended. && exit 2)
+	mkdir -p $(INSTALL_BIN)
 	for i in $(INSTALLED_PROGRAMS) ""; do test -z "$$i" || $(INSTALL) $$i $(INSTALL_BIN); done
 
 install: install_libs install_programs
