--- a/libsres/ns_parse.c
+++ b/libsres/ns_parse.c
@@ -15,6 +15,7 @@
  * OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  */
 #include "validator-internal.h"
+#include "validator/validator-compat.h"
 
 #include "res_support.h"
 #include "res_comp.h"
@@ -48,7 +49,10 @@ static void     setsection(ns_msg * msg,
 /*
  * These need to be in the same order as the nres.h:ns_flag enum. 
  */
-struct _ns_flagdata _ns_flagdata_flags[16] = {
+
+struct _ns_flagdata {  int mask, shift;  };
+
+struct _ns_flagdata _ns_flagdata_flags[] = {
     {0x8000, 15},               /* qr. */
     {0x7800, 11},               /* opcode. */
     {0x0400, 10},               /* aa. */
