diff --git a/libsres/ns_parse.c b/libsres/ns_parse.c
index a89358e..ef37ad2 100644
--- a/libsres/ns_parse.c
+++ b/libsres/ns_parse.c
@@ -16,6 +16,7 @@
  */
 #include "validator/validator-config.h"
 #include "validator-internal.h"
+#include "validator/validator-compat.h"
 
 #include "res_support.h"
 #include "res_comp.h"
@@ -49,7 +50,10 @@ static void     setsection(ns_msg * msg, ns_sect sect);
 /*
  * These need to be in the same order as the nres.h:ns_flag enum. 
  */
-struct _ns_flagdata _ns_flagdata_flags[16] = {
+
+struct _ns_flagdata {  int mask, shift;  };
+
+struct _ns_flagdata _ns_flagdata_flags[] = {
     {0x8000, 15},               /* qr. */
     {0x7800, 11},               /* opcode. */
     {0x0400, 10},               /* aa. */
