From 3dfcb762f12d8717148d9d5ec1d36ec97b7275a6 Mon Sep 17 00:00:00 2001
From: Dave Taht <dave.taht@bufferbloat.net>
Date: Thu, 16 Feb 2012 08:42:07 -0800
Subject: [PATCH 2/2] miniupnpc description change

---
 upnpcommands.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/upnpcommands.c b/upnpcommands.c
index 1114759..8e3ec5e 100644
--- a/upnpcommands.c
+++ b/upnpcommands.c
@@ -367,7 +367,7 @@ UPNP_AddPortMapping(const char * controlURL, const char * servicetype,
 	AddPortMappingArgs[5].elt = "NewEnabled";
 	AddPortMappingArgs[5].val = "1";
 	AddPortMappingArgs[6].elt = "NewPortMappingDescription";
-	AddPortMappingArgs[6].val = desc?desc:"libminiupnpc";
+	AddPortMappingArgs[6].val = desc?desc:"miniupnpc";
 	AddPortMappingArgs[7].elt = "NewLeaseDuration";
 	AddPortMappingArgs[7].val = leaseDuration?leaseDuration:"0";
 	if(!(buffer = simpleUPnPcommand(-1, controlURL, servicetype,
-- 
1.7.1

