--- a/configure.ac
+++ b/configure.ac
@@ -14,14 +14,6 @@ AC_PROG_INSTALL
 AC_PROG_MKDIR_P
 AC_LANG(C)
 
-if test -d "/usr/local/lib"; then
-   LDFLAGS="$LDFLAGS -L/usr/local/lib"
-fi
-
-if test -d "/usr/local/include"; then
-   CFLAGS="$CFLAGS -I/usr/local/include"
-fi
-
 # Whether all the debugging output should be spewed out
 AC_ARG_ENABLE([debug],
   [AS_HELP_STRING([--enable-debug], [build with debugging symbols])])
