*** samba-3.6.1.orig/source3/Makefile.in	2011-11-25 11:37:07.634171027 -0800
--- samba-3.6.1/source3/Makefile.in	2011-12-01 00:05:43.809459278 -0800
***************
*** 49,55 ****
  AR=@AR@
  LDSHFLAGS=$(PICFLAG) @LDSHFLAGS@ @RELRO_LDFLAGS@ @LDFLAGS@ -lc @LDSHFLAGS_Z_DEFS@
  LDSHFLAGS_MODULES=$(PICFLAG) @LDSHFLAGS@ @RELRO_LDFLAGS@ @LDFLAGS@ @LDSHFLAGS_Z_NODEFS@
! LDFLAGS=@PIE_LDFLAGS@ @RELRO_LDFLAGS@ @LDFLAGS@
  
  WINBIND_NSS_LDSHFLAGS=$(PICFLAG) @WINBIND_NSS_LDSHFLAGS@ @LDFLAGS@ -lc @LDSHFLAGS_Z_DEFS@
  AWK=@AWK@
--- 49,55 ----
  AR=@AR@
  LDSHFLAGS=$(PICFLAG) @LDSHFLAGS@ @RELRO_LDFLAGS@ @LDFLAGS@ -lc @LDSHFLAGS_Z_DEFS@
  LDSHFLAGS_MODULES=$(PICFLAG) @LDSHFLAGS@ @RELRO_LDFLAGS@ @LDFLAGS@ @LDSHFLAGS_Z_NODEFS@
! LDFLAGS=@RELRO_LDFLAGS@ @LDFLAGS@
  
  WINBIND_NSS_LDSHFLAGS=$(PICFLAG) @WINBIND_NSS_LDSHFLAGS@ @LDFLAGS@ -lc @LDSHFLAGS_Z_DEFS@
  AWK=@AWK@
***************
*** 85,92 ****
  LIBTDB=@LIBTDB_STATIC@ @LIBTDB_SHARED@
  LIBTDB_LIBS=@LIBTDB_LIBS@
  TDB_DEPS=@TDB_DEPS@
! LIBNETAPI=@LIBNETAPI_STATIC@ @LIBNETAPI_SHARED@
  LIBNETAPI_LIBS=@LIBNETAPI_LIBS@
  LIBSMBCLIENT_LIBS=@LIBSMBCLIENT_LIBS@
  LIBSMBSHAREMODES_LIBS=@LIBSMBSHAREMODES_LIBS@
  
--- 85,93 ----
  LIBTDB=@LIBTDB_STATIC@ @LIBTDB_SHARED@
  LIBTDB_LIBS=@LIBTDB_LIBS@
  TDB_DEPS=@TDB_DEPS@
! LIBNETAPI=@LIBNETAPI_STATIC@ # @LIBNETAPI_SHARED@
  LIBNETAPI_LIBS=@LIBNETAPI_LIBS@
+ LIBSMBCLIENT=@LIBSMBCLIENT_STATIC@ @LIBSMBCLIENT_SHARED@
  LIBSMBCLIENT_LIBS=@LIBSMBCLIENT_LIBS@
  LIBSMBSHAREMODES_LIBS=@LIBSMBSHAREMODES_LIBS@
  
***************
*** 268,279 ****
  # object file lists
  ######################################################################
  
! TDB_LIB_OBJ = lib/util_tdb.o ../lib/util/util_tdb.o \
! 	  ../lib/util/tdb_wrap.o \
! 	  lib/dbwrap.o lib/dbwrap_tdb.o \
! 	  lib/dbwrap_ctdb.o \
! 	  lib/g_lock.o \
! 	  lib/dbwrap_rbt.o
  
  TDB_VALIDATE_OBJ = lib/tdb_validate.o
  
--- 269,277 ----
  # object file lists
  ######################################################################
  
! TDB_LIB_OBJ = lib/util_tdb.o ../lib/util/util_tdb.o ../lib/util/tdb_wrap.o \
! 	      lib/dbwrap.o lib/dbwrap_tdb.o lib/dbwrap_ctdb.o lib/g_lock.o \
! 	      lib/dbwrap_rbt.o
  
  TDB_VALIDATE_OBJ = lib/tdb_validate.o
  
***************
*** 302,309 ****
  ZLIB_OBJ = @ZLIB_OBJS@
  
  COMPRESSION_OBJ = ../lib/compression/mszip.o \
! 				  ../lib/compression/lzxpress.o \
! 				  $(ZLIB_OBJ)
  
  DRSUAPI_OBJ = $(LIBNDR_DRSUAPI_OBJ) \
  	      $(COMPRESSION_OBJ)
--- 300,307 ----
  ZLIB_OBJ = @ZLIB_OBJS@
  
  COMPRESSION_OBJ = ../lib/compression/mszip.o \
! 		  ../lib/compression/lzxpress.o \
! 		  $(ZLIB_OBJ)
  
  DRSUAPI_OBJ = $(LIBNDR_DRSUAPI_OBJ) \
  	      $(COMPRESSION_OBJ)
***************
*** 409,440 ****
  
  LIBREPLACE_OBJ = @LIBREPLACE_OBJS@
  
- SOCKET_WRAPPER_OBJ = @SOCKET_WRAPPER_OBJS@
  NSS_WRAPPER_OBJ = @NSS_WRAPPER_OBJS@
  
! LIBSAMBAUTIL_OBJ = \
! 		$(LIBREPLACE_OBJ) \
! 		$(SOCKET_WRAPPER_OBJ) \
! 		$(NSS_WRAPPER_OBJ)
  
  UTIL_OBJ = ../lib/util/rbtree.o ../lib/util/signal.o ../lib/util/time.o \
! 		   ../lib/util/xfile.o ../lib/util/util_strlist.o  \
! 		   ../lib/util/util_file.o ../lib/util/data_blob.o \
! 		   ../lib/util/util.o ../lib/util/fsusage.o \
! 		   ../lib/util/params.o ../lib/util/talloc_stack.o \
! 		   ../lib/util/genrand.o ../lib/util/util_net.o \
! 		   ../lib/util/become_daemon.o ../lib/util/system.o \
! 		   ../lib/util/tevent_unix.o ../lib/util/tevent_ntstatus.o \
! 		   ../lib/util/tevent_werror.o \
! 		   ../lib/util/smb_threads.o ../lib/util/util_id.o \
! 		   ../lib/util/blocking.o ../lib/util/rfc1738.o \
! 		   ../lib/util/select.o ../lib/util/util_pw.o
! 
! CRYPTO_OBJ = ../lib/crypto/crc32.o ../lib/crypto/md5.o \
! 			 ../lib/crypto/hmacmd5.o ../lib/crypto/arcfour.o \
! 			 ../lib/crypto/md4.o \
! 			 ../lib/crypto/sha256.o ../lib/crypto/hmacsha256.o \
! 			 ../lib/crypto/aes.o ../lib/crypto/rijndael-alg-fst.o
  
  LIB_OBJ = $(LIBSAMBAUTIL_OBJ) $(UTIL_OBJ) $(CRYPTO_OBJ) \
  	  lib/messages.o librpc/gen_ndr/ndr_messaging.o lib/messages_local.o \
--- 407,437 ----
  
  LIBREPLACE_OBJ = @LIBREPLACE_OBJS@
  
  NSS_WRAPPER_OBJ = @NSS_WRAPPER_OBJS@
  
! SOCKET_WRAPPER_OBJ = @SOCKET_WRAPPER_OBJS@
! 
! LIBSAMBAUTIL_OBJ = $(LIBREPLACE_OBJ) \
! 		   $(SOCKET_WRAPPER_OBJ) \
! 		   $(NSS_WRAPPER_OBJ)
  
  UTIL_OBJ = ../lib/util/rbtree.o ../lib/util/signal.o ../lib/util/time.o \
! 	   ../lib/util/xfile.o ../lib/util/util_strlist.o  \
! 	   ../lib/util/util_file.o ../lib/util/data_blob.o \
! 	   ../lib/util/util.o ../lib/util/fsusage.o \
! 	   ../lib/util/params.o ../lib/util/talloc_stack.o \
! 	   ../lib/util/genrand.o ../lib/util/util_net.o \
! 	   ../lib/util/become_daemon.o ../lib/util/system.o \
! 	   ../lib/util/tevent_unix.o ../lib/util/tevent_ntstatus.o \
! 	   ../lib/util/tevent_werror.o \
! 	   ../lib/util/smb_threads.o ../lib/util/util_id.o \
! 	   ../lib/util/blocking.o ../lib/util/rfc1738.o \
! 	   ../lib/util/select.o ../lib/util/util_pw.o
! 
! CRYPTO_OBJ = ../lib/crypto/crc32.o ../lib/crypto/md5.o ../lib/crypto/hmacmd5.o \
! 	     ../lib/crypto/arcfour.o ../lib/crypto/md4.o \
! 	     ../lib/crypto/sha256.o ../lib/crypto/hmacsha256.o \
! 	     ../lib/crypto/aes.o ../lib/crypto/rijndael-alg-fst.o
  
  LIB_OBJ = $(LIBSAMBAUTIL_OBJ) $(UTIL_OBJ) $(CRYPTO_OBJ) \
  	  lib/messages.o librpc/gen_ndr/ndr_messaging.o lib/messages_local.o \
***************
*** 446,452 ****
  	  lib/util_transfer_file.o ../lib/async_req/async_sock.o \
  	  lib/addrchange.o \
  	  $(TDB_LIB_OBJ) \
! 	  $(VERSION_OBJ) lib/charcnv.o ../lib/util/debug.o ../lib/util/debug_s3.o lib/fault.o \
  	  lib/interface.o lib/pidfile.o \
  	  lib/system.o lib/sendfile.o lib/recvfile.o lib/time.o \
  	  lib/username.o \
--- 443,450 ----
  	  lib/util_transfer_file.o ../lib/async_req/async_sock.o \
  	  lib/addrchange.o \
  	  $(TDB_LIB_OBJ) \
! 	  $(VERSION_OBJ) lib/charcnv.o \
! 	  ../lib/util/debug.o ../lib/util/debug_s3.o lib/fault.o \
  	  lib/interface.o lib/pidfile.o \
  	  lib/system.o lib/sendfile.o lib/recvfile.o lib/time.o \
  	  lib/username.o \
***************
*** 487,509 ****
  # Be sure to include them into your application
  POPT_LIB_OBJ = lib/popt_common.o
  
! PARAM_WITHOUT_REG_OBJ = dynconfig.o param/loadparm.o param/loadparm_server_role.o param/util.o lib/sharesec.o lib/ldap_debug_handler.o
  PARAM_REG_ADD_OBJ = $(REG_SMBCONF_OBJ) $(LIBSMBCONF_OBJ) $(PRIVILEGES_BASIC_OBJ)
  PARAM_OBJ = $(PARAM_WITHOUT_REG_OBJ) $(PARAM_REG_ADD_OBJ)
  
  KRBCLIENT_OBJ = libads/kerberos.o libads/ads_status.o
  
! LIBADDNS_OBJ0 = ../lib/addns/dnsrecord.o ../lib/addns/dnsutils.o  ../lib/addns/dnssock.o \
! 	       ../lib/addns/dnsgss.o ../lib/addns/dnsmarshall.o ../lib/addns/error.o
  LIBADDNS_OBJ = $(LIBADDNS_OBJ0) $(SOCKET_WRAPPER_OBJ)
  
  GPEXT_OBJ = ../libgpo/gpext/gpext.o @GPEXT_STATIC@
  
! LIBGPO_OBJ0 = ../libgpo/gpo_ldap.o ../libgpo/gpo_ini.o ../libgpo/gpo_util.o \
! 	      ../libgpo/gpo_fetch.o libgpo/gpo_filesync.o ../libgpo/gpo_sec.o \
! 	      libgpo/gpo_reg.o \
! 	      $(GPEXT_OBJ)
! LIBGPO_OBJ = $(LIBGPO_OBJ0)
  
  LIBADS_OBJ = libads/ldap.o \
  	     libads/sasl.o libads/sasl_wrapping.o \
--- 485,510 ----
  # Be sure to include them into your application
  POPT_LIB_OBJ = lib/popt_common.o
  
! PARAM_WITHOUT_REG_OBJ = dynconfig.o param/loadparm.o \
! 			param/loadparm_server_role.o param/util.o \
! 			lib/sharesec.o lib/ldap_debug_handler.o
  PARAM_REG_ADD_OBJ = $(REG_SMBCONF_OBJ) $(LIBSMBCONF_OBJ) $(PRIVILEGES_BASIC_OBJ)
  PARAM_OBJ = $(PARAM_WITHOUT_REG_OBJ) $(PARAM_REG_ADD_OBJ)
  
  KRBCLIENT_OBJ = libads/kerberos.o libads/ads_status.o
  
! LIBADDNS_OBJ0 = ../lib/addns/dnsrecord.o ../lib/addns/dnsutils.o \
! 		../lib/addns/dnssock.o ../lib/addns/dnsgss.o \
! 		../lib/addns/dnsmarshall.o ../lib/addns/error.o
! 
  LIBADDNS_OBJ = $(LIBADDNS_OBJ0) $(SOCKET_WRAPPER_OBJ)
  
  GPEXT_OBJ = ../libgpo/gpext/gpext.o @GPEXT_STATIC@
  
! LIBGPO_OBJ = ../libgpo/gpo_ldap.o ../libgpo/gpo_ini.o ../libgpo/gpo_util.o \
! 	     ../libgpo/gpo_fetch.o libgpo/gpo_filesync.o ../libgpo/gpo_sec.o \
! 	     libgpo/gpo_reg.o \
! 	     $(GPEXT_OBJ)
  
  LIBADS_OBJ = libads/ldap.o \
  	     libads/sasl.o libads/sasl_wrapping.o \
***************
*** 546,553 ****
  LIBSMB_ERR_OBJ = $(LIBSMB_ERR_OBJ0) $(LIBSMB_ERR_OBJ1) \
  		 $(SECRETS_OBJ)
  
! LIBSMB_OBJ0 = \
! 	       ../libcli/auth/ntlm_check.o \
  	       libsmb/ntlmssp.o \
  	       libsmb/ntlmssp_wrap.o \
  	       ../libcli/auth/ntlmssp.o \
--- 547,553 ----
  LIBSMB_ERR_OBJ = $(LIBSMB_ERR_OBJ0) $(LIBSMB_ERR_OBJ1) \
  		 $(SECRETS_OBJ)
  
! LIBSMB_OBJ0 = ../libcli/auth/ntlm_check.o \
  	       libsmb/ntlmssp.o \
  	       libsmb/ntlmssp_wrap.o \
  	       ../libcli/auth/ntlmssp.o \
***************
*** 563,575 ****
  LIBCLI_LDAP_NDR_OBJ = ../libcli/ldap/ldap_ndr.o lib/ldb_compat.o
  
  LIBTSOCKET_OBJ = ../lib/tsocket/tsocket.o \
! 		../lib/tsocket/tsocket_helpers.o \
! 		../lib/tsocket/tsocket_bsd.o
  
  CLDAP_OBJ = libads/cldap.o \
! 	../libcli/cldap/cldap.o \
! 	../lib/util/idtree.o \
! 	$(LIBCLI_LDAP_MESSAGE_OBJ) $(LIBCLI_LDAP_NDR_OBJ) $(LIBTSOCKET_OBJ)
  
  TLDAP_OBJ = lib/tldap.o lib/tldap_util.o lib/util_tsock.o
  
--- 563,575 ----
  LIBCLI_LDAP_NDR_OBJ = ../libcli/ldap/ldap_ndr.o lib/ldb_compat.o
  
  LIBTSOCKET_OBJ = ../lib/tsocket/tsocket.o \
! 		 ../lib/tsocket/tsocket_helpers.o \
! 		 ../lib/tsocket/tsocket_bsd.o
  
  CLDAP_OBJ = libads/cldap.o \
! 	    ../libcli/cldap/cldap.o \
! 	    ../lib/util/idtree.o \
! 	    $(LIBCLI_LDAP_MESSAGE_OBJ) $(LIBCLI_LDAP_NDR_OBJ) $(LIBTSOCKET_OBJ)
  
  TLDAP_OBJ = lib/tldap.o lib/tldap_util.o lib/util_tsock.o
  
***************
*** 657,664 ****
  		  $(REG_BACKENDS_SMBCONF_OBJ) \
  		  $(REG_INIT_SMBCONF_OBJ)
  
! REG_FULL_OBJ = $(REG_SMBCONF_OBJ) \
! 	       $(REG_BACKENDS_EXTRA_OBJ) \
  	       $(REG_INIT_FULL_OBJ) \
  	       registry/reg_perfcount.o \
  	       librpc/gen_ndr/ndr_perfcount.o
--- 657,663 ----
  		  $(REG_BACKENDS_SMBCONF_OBJ) \
  		  $(REG_INIT_SMBCONF_OBJ)
  
! REG_FULL_OBJ = $(REG_BACKENDS_EXTRA_OBJ) \
  	       $(REG_INIT_FULL_OBJ) \
  	       registry/reg_perfcount.o \
  	       librpc/gen_ndr/ndr_perfcount.o
***************
*** 678,709 ****
                 rpc_server/samr/srv_samr_chgpasswd.o \
  	       librpc/gen_ndr/srv_samr.o
  
! RPC_INITSHUTDOWN_OBJ =  librpc/gen_ndr/srv_initshutdown.o rpc_server/initshutdown/srv_initshutdown_nt.o
  
! RPC_WINREG_OBJ =  rpc_server/winreg/srv_winreg_nt.o \
! 		  librpc/gen_ndr/srv_winreg.o
  
! RPC_DSSETUP_OBJ =  rpc_server/dssetup/srv_dssetup_nt.o librpc/gen_ndr/srv_dssetup.o
  
  RPC_SRVSVC_OBJ = rpc_server/srvsvc/srv_srvsvc_nt.o \
  		 librpc/gen_ndr/srv_srvsvc.o
  
! RPC_WKSSVC_OBJ =  librpc/gen_ndr/srv_wkssvc.o \
! 		  rpc_server/wkssvc/srv_wkssvc_nt.o
  
! RPC_SVCCTL_OBJ =  rpc_server/svcctl/srv_svcctl_nt.o \
! 		  rpc_server/svcctl/srv_svcctl_reg.o \
! 		  librpc/gen_ndr/srv_svcctl.o \
! 		  services/svc_winreg_glue.o \
!                   services/svc_spoolss.o services/svc_rcinit.o  \
!                   services/svc_netlogon.o services/svc_winreg.o \
!                   services/svc_wins.o
  
  RPC_NTSVCS_OBJ = rpc_server/ntsvcs/srv_ntsvcs_nt.o \
  		 librpc/gen_ndr/srv_ntsvcs.o
  
! RPC_NETDFS_OBJ =  librpc/gen_ndr/srv_dfs.o \
! 		  rpc_server/dfs/srv_dfs_nt.o
  
  RPC_SPOOLSS_OBJ = rpc_server/spoolss/srv_spoolss_nt.o \
  		  rpc_server/spoolss/srv_spoolss_util.o \
--- 677,708 ----
                 rpc_server/samr/srv_samr_chgpasswd.o \
  	       librpc/gen_ndr/srv_samr.o
  
! RPC_INITSHUTDOWN_OBJ = librpc/gen_ndr/srv_initshutdown.o \
! 		       rpc_server/initshutdown/srv_initshutdown_nt.o
  
! RPC_WINREG_OBJ = rpc_server/winreg/srv_winreg_nt.o librpc/gen_ndr/srv_winreg.o
  
! RPC_DSSETUP_OBJ = rpc_server/dssetup/srv_dssetup_nt.o librpc/gen_ndr/srv_dssetup.o
  
  RPC_SRVSVC_OBJ = rpc_server/srvsvc/srv_srvsvc_nt.o \
  		 librpc/gen_ndr/srv_srvsvc.o
  
! RPC_WKSSVC_OBJ = librpc/gen_ndr/srv_wkssvc.o \
! 		 rpc_server/wkssvc/srv_wkssvc_nt.o
  
! RPC_SVCCTL_OBJ = rpc_server/svcctl/srv_svcctl_nt.o \
! 		 rpc_server/svcctl/srv_svcctl_reg.o \
! 		 librpc/gen_ndr/srv_svcctl.o \
! 		 services/svc_winreg_glue.o \
!                  services/svc_spoolss.o services/svc_rcinit.o  \
!                  services/svc_netlogon.o services/svc_winreg.o \
!                  services/svc_wins.o
  
  RPC_NTSVCS_OBJ = rpc_server/ntsvcs/srv_ntsvcs_nt.o \
  		 librpc/gen_ndr/srv_ntsvcs.o
  
! RPC_NETDFS_OBJ = librpc/gen_ndr/srv_dfs.o \
! 		 rpc_server/dfs/srv_dfs_nt.o
  
  RPC_SPOOLSS_OBJ = rpc_server/spoolss/srv_spoolss_nt.o \
  		  rpc_server/spoolss/srv_spoolss_util.o \
***************
*** 723,730 ****
  RPC_SERVICE = rpc_server/rpc_server.o
  
  RPC_CRYPTO = rpc_server/dcesrv_ntlmssp.o \
! 		rpc_server/dcesrv_gssapi.o \
! 		rpc_server/dcesrv_spnego.o
  
  RPC_PIPE_OBJ = rpc_server/srv_pipe.o rpc_server/srv_pipe_hnd.o \
  	       $(RPC_NCACN_NP) $(RPC_SERVICE) $(RPC_CRYPTO)
--- 722,729 ----
  RPC_SERVICE = rpc_server/rpc_server.o
  
  RPC_CRYPTO = rpc_server/dcesrv_ntlmssp.o \
! 	     rpc_server/dcesrv_gssapi.o \
! 	     rpc_server/dcesrv_spnego.o
  
  RPC_PIPE_OBJ = rpc_server/srv_pipe.o rpc_server/srv_pipe_hnd.o \
  	       $(RPC_NCACN_NP) $(RPC_SERVICE) $(RPC_CRYPTO)
***************
*** 735,756 ****
  
  RPC_SERVER_REGISTER_OBJ = rpc_server/rpc_ep_setup.o $(DCE_RPC_EP_OBJ)
  
! RPC_SERVER_OBJ = $(RPC_LSARPC_OBJ) $(RPC_WINREG_OBJ) $(RPC_INITSHUTDOWN_OBJ) \
  		 $(RPC_DSSETUP_OBJ) $(RPC_WKSSVC_OBJ) $(RPC_SVCCTL_OBJ) \
  		 $(RPC_NTSVCS_OBJ) $(RPC_NETLOGON_OBJ) $(RPC_NETDFS_OBJ) \
  		 $(RPC_SRVSVC_OBJ) $(RPC_SPOOLSS_OBJ) $(RPC_EVENTLOG_OBJ) \
  		 $(RPC_SAMR_OBJ) $(RPC_RPCECHO_OBJ) $(RPC_EPMAPPER_OBJ) \
! 		 $(RPC_PIPE_OBJ) $(NPA_TSTREAM_OBJ) \
! 		 $(LIBCLI_SPOOLSS_OBJ) \
! 		 $(LIBCLI_WINREG_INT_OBJ) \
! 		 $(LIBCLI_SRVSVC_OBJ) \
! 		 $(LIBCLI_LSA_OBJ) \
! 		 $(LIBCLI_SAMR_OBJ) \
! 		 $(LIBCLI_NETLOGON_OBJ) \
! 		 $(RPC_SERVER_REGISTER_OBJ) \
! 		 $(RPC_CLIENT_SCHANNEL_OBJ) \
! 		 rpc_client/init_netlogon.o \
! 		 rpc_client/init_samr.o
  
  RPC_CLIENT_SCHANNEL_OBJ = rpc_client/cli_pipe_schannel.o
  
--- 734,748 ----
  
  RPC_SERVER_REGISTER_OBJ = rpc_server/rpc_ep_setup.o $(DCE_RPC_EP_OBJ)
  
! RPC_SERVER_OBJ = $(LIBCLI_SPOOLSS_OBJ) $(LIBCLI_WINREG_INT_OBJ) \
! 		 $(NPA_TSTREAM_OBJ) \
! 		 $(RPC_LSARPC_OBJ) $(RPC_WINREG_OBJ) $(RPC_INITSHUTDOWN_OBJ) \
  		 $(RPC_DSSETUP_OBJ) $(RPC_WKSSVC_OBJ) $(RPC_SVCCTL_OBJ) \
  		 $(RPC_NTSVCS_OBJ) $(RPC_NETLOGON_OBJ) $(RPC_NETDFS_OBJ) \
  		 $(RPC_SRVSVC_OBJ) $(RPC_SPOOLSS_OBJ) $(RPC_EVENTLOG_OBJ) \
  		 $(RPC_SAMR_OBJ) $(RPC_RPCECHO_OBJ) $(RPC_EPMAPPER_OBJ) \
! 		 $(RPC_PIPE_OBJ) $(RPC_CLIENT_SCHANNEL_OBJ) \
! 		 $(RPC_SERVER_REGISTER_OBJ)
  
  RPC_CLIENT_SCHANNEL_OBJ = rpc_client/cli_pipe_schannel.o
  
***************
*** 762,774 ****
  
  PASSDB_GET_SET_OBJ = passdb/pdb_get_set.o
  
! PASSDB_OBJ = $(PASSDB_GET_SET_OBJ) passdb/passdb.o passdb/pdb_interface.o \
! 		lib/util_wellknown.o lib/util_builtin.o passdb/pdb_compat.o \
! 		lib/util_unixsids.o passdb/lookup_sid.o \
! 		passdb/login_cache.o @PDB_STATIC@ \
! 		passdb/account_pol.o $(PRIVILEGES_OBJ) \
! 		lib/util_nscd.o lib/winbind_util.o $(SERVER_MUTEX_OBJ) \
! 		passdb/pdb_util.o
  
  DEVEL_HELP_WEIRD_OBJ = modules/weird.o
  CP850_OBJ = modules/CP850.o
--- 754,765 ----
  
  PASSDB_GET_SET_OBJ = passdb/pdb_get_set.o
  
! PASSDB_OBJ = lib/util_builtin.o lib/util_nscd.o lib/util_unixsids.o \
! 	     lib/util_wellknown.o lib/winbind_util.o \
! 	     passdb/account_pol.o passdb/login_cache.o passdb/lookup_sid.o \
! 	     passdb/passdb.o passdb/pdb_compat.o passdb/pdb_interface.o \
! 	     passdb/pdb_util.o $(PASSDB_GET_SET_OBJ) @PDB_STATIC@ \
! 	     $(PRIVILEGES_OBJ) $(SERVER_MUTEX_OBJ)
  
  DEVEL_HELP_WEIRD_OBJ = modules/weird.o
  CP850_OBJ = modules/CP850.o
***************
*** 778,789 ****
  GROUPDB_OBJ = groupdb/mapping.o groupdb/mapping_tdb.o
  
  PROFILE_OBJ = profile/profile.o
! PROFILES_OBJ = utils/profiles.o \
! 	       $(LIBSMB_ERR_OBJ) \
! 	       $(PARAM_OBJ) \
!                $(LIB_OBJ) $(LIB_DUMMY_OBJ) \
!                $(POPT_LIB_OBJ) \
! 	       $(REGFIO_OBJ)
  
  OPLOCK_OBJ = smbd/oplock.o smbd/oplock_irix.o smbd/oplock_linux.o \
  	     smbd/oplock_onefs.o
--- 769,776 ----
  GROUPDB_OBJ = groupdb/mapping.o groupdb/mapping_tdb.o
  
  PROFILE_OBJ = profile/profile.o
! 
! PROFILES_OBJ = utils/profiles.o
  
  OPLOCK_OBJ = smbd/oplock.o smbd/oplock_irix.o smbd/oplock_linux.o \
  	     smbd/oplock_onefs.o
***************
*** 851,857 ****
  
  SLCACHE_OBJ = libsmb/samlogon_cache.o
  
! DCUTIL_OBJ  = libsmb/namequery_dc.o libsmb/trustdom_cache.o libsmb/trusts_util.o libsmb/dsgetdcname.o
  
  AUTH_BUILTIN_OBJ = auth/auth_builtin.o
  AUTH_DOMAIN_OBJ = auth/auth_domain.o
--- 838,845 ----
  
  SLCACHE_OBJ = libsmb/samlogon_cache.o
  
! DCUTIL_OBJ = libsmb/dsgetdcname.o libsmb/namequery_dc.o \
! 	     libsmb/trustdom_cache.o libsmb/trusts_util.o
  
  AUTH_BUILTIN_OBJ = auth/auth_builtin.o
  AUTH_DOMAIN_OBJ = auth/auth_domain.o
***************
*** 870,904 ****
  	   auth/user_util.o \
  	   auth/user_krb5.o \
  	   auth/auth_compat.o auth/auth_ntlmssp.o \
! 	   $(PLAINTEXT_AUTH_OBJ) $(SLCACHE_OBJ) $(DCUTIL_OBJ)
  
  MANGLE_OBJ = smbd/mangle.o smbd/mangle_hash.o smbd/mangle_hash2.o
  
! SMBD_OBJ_MAIN = smbd/server.o smbd/server_exit.o smbd/msg_idmap.o
  
! BUILDOPT_OBJ = smbd/build_options.o
  
  SMBD_OBJ_SRV = smbd/server_reload.o \
  	       smbd/files.o smbd/connection.o \
  	       smbd/utmp.o smbd/session.o \
!                smbd/dfree.o smbd/dir.o smbd/password.o smbd/conn.o \
  	       smbd/share_access.o smbd/fileio.o \
!                smbd/ipc.o smbd/lanman.o smbd/negprot.o \
!                smbd/message.o smbd/nttrans.o smbd/pipes.o \
!                smbd/reply.o smbd/sesssetup.o smbd/trans2.o smbd/uid.o \
  	       smbd/dosmode.o smbd/filename.o smbd/open.o smbd/close.o \
  	       smbd/blocking.o smbd/sec_ctx.o smbd/srvstr.o \
  	       smbd/vfs.o smbd/perfcount.o smbd/statcache.o smbd/seal.o \
!                smbd/posix_acls.o lib/sysacls.o \
  	       smbd/process.o smbd/service.o smbd/error.o \
! 	       rpc_server/epmd.o \
  	       printing/printspoolss.o printing/spoolssd.o \
! 	       lib/sysquotas.o lib/sysquotas_linux.o \
! 	       lib/sysquotas_xfs.o lib/sysquotas_4A.o \
! 	       lib/sysquotas_nfs.o \
  	       smbd/fake_file.o \
! 	       smbd/quotas.o smbd/ntquotas.o $(AFS_OBJ) smbd/msdfs.o \
! 	       $(AFS_SETTOKEN_OBJ) smbd/aio.o smbd/statvfs.o \
  	       smbd/dmapi.o smbd/signing.o \
  	       smbd/file_access.o \
  	       smbd/dnsregister.o smbd/globals.o \
--- 858,929 ----
  	   auth/user_util.o \
  	   auth/user_krb5.o \
  	   auth/auth_compat.o auth/auth_ntlmssp.o \
! 	   $(PLAINTEXT_AUTH_OBJ) $(SLCACHE_OBJ)
! 
! BUILDOPT_OBJ = smbd/build_options.o
  
  MANGLE_OBJ = smbd/mangle.o smbd/mangle_hash.o smbd/mangle_hash2.o
  
! PRINTING_OBJ = printing/pcap.o printing/print_svid.o printing/print_aix.o \
!                printing/print_cups.o printing/print_generic.o \
!                printing/lpq_parse.o printing/load.o \
!                printing/print_iprint.o printing/print_standard.o \
!                printing/printer_list.o librpc/gen_ndr/ndr_printcap.o
  
! PRINTBASE_OBJ = printing/notify.o printing/printing_db.o
! 
! PRINTBACKEND_OBJ = printing/printing.o \
! 		   printing/nt_printing.o \
! 		   printing/nt_printing_tdb.o \
! 		   printing/nt_printing_migrate.o \
! 		   printing/nt_printing_migrate_internal.o \
! 		   printing/nt_printing_ads.o \
! 		   librpc/gen_ndr/ndr_ntprinting.o \
! 		    ../librpc/ndr/ndr_ntprinting.o \
! 		   $(PRINTBASE_OBJ)
! 
! NMBD_OBJ = nmbd/asyncdns.o nmbd/nmbd.o nmbd/nmbd_become_dmb.o \
! 	   nmbd/nmbd_become_lmb.o nmbd/nmbd_browserdb.o \
! 	   nmbd/nmbd_browsesync.o nmbd/nmbd_elections.o \
! 	   nmbd/nmbd_incomingdgrams.o nmbd/nmbd_incomingrequests.o \
! 	   nmbd/nmbd_lmhosts.o nmbd/nmbd_logonnames.o nmbd/nmbd_mynames.o \
! 	   nmbd/nmbd_namelistdb.o nmbd/nmbd_namequery.o \
! 	   nmbd/nmbd_nameregister.o nmbd/nmbd_namerelease.o \
! 	   nmbd/nmbd_nodestatus.o nmbd/nmbd_packets.o \
! 	   nmbd/nmbd_processlogon.o nmbd/nmbd_responserecordsdb.o \
! 	   nmbd/nmbd_sendannounce.o nmbd/nmbd_serverlistdb.o \
! 	   nmbd/nmbd_subnetdb.o nmbd/nmbd_winsproxy.o nmbd/nmbd_winsserver.o \
! 	   nmbd/nmbd_workgroupdb.o nmbd/nmbd_synclists.o
! 
! SMBCACLS_OBJ = utils/smbcacls.o
! 
! SMBCONTROL_OBJ = utils/smbcontrol.o \
! 		 $(PRINTBASE_OBJ)
! 
! SMBCQUOTAS_OBJ = utils/smbcquotas.o
! 
! SMBD_OBJ_MAIN = smbd/server.o smbd/server_exit.o smbd/msg_idmap.o
  
  SMBD_OBJ_SRV = smbd/server_reload.o \
  	       smbd/files.o smbd/connection.o \
  	       smbd/utmp.o smbd/session.o \
! 	       smbd/dfree.o smbd/dir.o smbd/password.o smbd/conn.o \
  	       smbd/share_access.o smbd/fileio.o \
! 	       smbd/ipc.o smbd/lanman.o smbd/negprot.o \
! 	       smbd/message.o smbd/nttrans.o smbd/pipes.o \
! 	       smbd/reply.o smbd/sesssetup.o smbd/trans2.o smbd/uid.o \
  	       smbd/dosmode.o smbd/filename.o smbd/open.o smbd/close.o \
  	       smbd/blocking.o smbd/sec_ctx.o smbd/srvstr.o \
  	       smbd/vfs.o smbd/perfcount.o smbd/statcache.o smbd/seal.o \
! 	       smbd/posix_acls.o lib/sysacls.o \
  	       smbd/process.o smbd/service.o smbd/error.o \
! 	       lib/sysquotas.o lib/sysquotas_linux.o lib/sysquotas_xfs.o \
! 	       lib/sysquotas_4A.o lib/sysquotas_nfs.o \
  	       printing/printspoolss.o printing/spoolssd.o \
! 	       rpc_server/epmd.o \
  	       smbd/fake_file.o \
! 	       smbd/quotas.o smbd/ntquotas.o smbd/msdfs.o \
! 	       $(AFS_OBJ) $(AFS_SETTOKEN_OBJ) smbd/aio.o smbd/statvfs.o \
  	       smbd/dmapi.o smbd/signing.o \
  	       smbd/file_access.o \
  	       smbd/dnsregister.o smbd/globals.o \
***************
*** 924,1040 ****
  	       ../libcli/smb/smb2_create_blob.o \
  	       $(MANGLE_OBJ) @VFS_STATIC@
  
! SMBD_OBJ_BASE = $(PARAM_WITHOUT_REG_OBJ) $(SMBD_OBJ_SRV) $(LIBSMB_OBJ) \
! 		$(RPC_SERVER_OBJ) \
! 		$(LOCKING_OBJ) $(PASSDB_OBJ) $(PRINTING_OBJ) $(PROFILE_OBJ) \
! 		$(LIB_OBJ) $(PRINTBACKEND_OBJ) $(OPLOCK_OBJ) \
! 		$(NOTIFY_OBJ) $(FNAME_UTIL_OBJ) $(GROUPDB_OBJ) $(AUTH_OBJ) \
! 		$(LIBMSRPC_OBJ) $(LIBMSRPC_GEN_OBJ) $(AVAHI_OBJ) \
! 		$(LIBADS_OBJ) $(KRBCLIENT_OBJ) $(LIBADS_SERVER_OBJ) $(LIBADS_PRINTER_OBJ) \
! 		$(REG_FULL_OBJ) $(POPT_LIB_OBJ) $(BUILDOPT_OBJ) \
! 		$(SMBLDAP_OBJ) $(LIBNET_OBJ) \
! 		$(LIBSMBCONF_OBJ) \
! 		$(PRIVILEGES_BASIC_OBJ) \
! 		$(REGFIO_OBJ) \
! 		$(REG_API_REGF_OBJ) \
! 		$(LIBNDR_XATTR_OBJ)
  
! PRINTING_OBJ = printing/pcap.o printing/print_svid.o printing/print_aix.o \
!                printing/print_cups.o printing/print_generic.o \
!                printing/lpq_parse.o printing/load.o \
!                printing/print_iprint.o printing/print_standard.o \
!                printing/printer_list.o librpc/gen_ndr/ndr_printcap.o
  
! PRINTBASE_OBJ = printing/notify.o printing/printing_db.o
! PRINTBACKEND_OBJ = printing/printing.o \
! 		   printing/nt_printing.o \
! 		   printing/nt_printing_tdb.o \
! 		   printing/nt_printing_migrate.o \
! 		   printing/nt_printing_migrate_internal.o \
! 		   printing/nt_printing_ads.o \
! 		   librpc/gen_ndr/ndr_ntprinting.o \
! 		    ../librpc/ndr/ndr_ntprinting.o \
! 		   $(PRINTBASE_OBJ)
  
! SMBD_OBJ = $(SMBD_OBJ_BASE) $(SMBD_OBJ_MAIN)
  
! NMBD_OBJ1 = nmbd/asyncdns.o nmbd/nmbd.o nmbd/nmbd_become_dmb.o \
!             nmbd/nmbd_become_lmb.o nmbd/nmbd_browserdb.o \
!             nmbd/nmbd_browsesync.o nmbd/nmbd_elections.o \
!             nmbd/nmbd_incomingdgrams.o nmbd/nmbd_incomingrequests.o \
!             nmbd/nmbd_lmhosts.o nmbd/nmbd_logonnames.o nmbd/nmbd_mynames.o \
!             nmbd/nmbd_namelistdb.o nmbd/nmbd_namequery.o \
!             nmbd/nmbd_nameregister.o nmbd/nmbd_namerelease.o \
!             nmbd/nmbd_nodestatus.o nmbd/nmbd_packets.o \
!             nmbd/nmbd_processlogon.o nmbd/nmbd_responserecordsdb.o \
!             nmbd/nmbd_sendannounce.o nmbd/nmbd_serverlistdb.o \
!             nmbd/nmbd_subnetdb.o nmbd/nmbd_winsproxy.o nmbd/nmbd_winsserver.o \
!             nmbd/nmbd_workgroupdb.o nmbd/nmbd_synclists.o
! 
! NMBD_OBJ = $(NMBD_OBJ1) $(PARAM_OBJ) $(LIBSMB_OBJ) $(KRBCLIENT_OBJ) \
!            $(PROFILE_OBJ) $(LIB_NONSMBD_OBJ) $(POPT_LIB_OBJ) \
! 	   $(LIBNDR_GEN_OBJ0)
! 
! SWAT_OBJ1 = web/cgi.o web/diagnose.o web/startstop.o web/statuspage.o \
!            web/swat.o web/neg_lang.o
! 
! SWAT_OBJ = $(SWAT_OBJ1) $(PARAM_OBJ) $(PRINTING_OBJ) $(PRINTBASE_OBJ) $(LIBSMB_OBJ) \
! 	   $(LOCKING_OBJ) $(PASSDB_OBJ) $(KRBCLIENT_OBJ) \
! 	   $(LIB_NONSMBD_OBJ) $(GROUPDB_OBJ) $(PLAINTEXT_AUTH_OBJ) \
! 	   $(POPT_LIB_OBJ) $(SMBLDAP_OBJ) $(LIBMSRPC_GEN_OBJ) $(LIBMSRPC_OBJ) \
!            $(PASSCHANGE_OBJ) $(FNAME_UTIL_OBJ) \
! 	   $(LIBCLI_SAMR_OBJ) \
! 	   rpc_client/init_lsa.o
  
  STATUS_OBJ = utils/status.o utils/status_profile.o \
! 	     $(LOCKING_OBJ) $(PARAM_OBJ) \
!              $(PROFILE_OBJ) $(LIB_NONSMBD_OBJ) $(POPT_LIB_OBJ) \
! 	     $(LIBSMB_ERR_OBJ) $(FNAME_UTIL_OBJ)
! 
! SMBCONTROL_OBJ = utils/smbcontrol.o $(PARAM_OBJ) $(LIB_NONSMBD_OBJ) \
! 	$(LIBSMB_ERR_OBJ) $(POPT_LIB_OBJ) $(PRINTBASE_OBJ)
! 
! SMBTREE_OBJ = utils/smbtree.o $(PARAM_OBJ) \
!              $(PROFILE_OBJ) $(LIB_NONSMBD_OBJ) $(LIBSMB_OBJ) \
! 	     $(KRBCLIENT_OBJ) $(POPT_LIB_OBJ) \
! 	     $(PASSDB_OBJ) $(SMBLDAP_OBJ) $(GROUPDB_OBJ) \
! 	     $(LIBMSRPC_GEN_OBJ) \
! 	     $(LIBMSRPC_OBJ) \
! 	     $(LIBCLI_SRVSVC_OBJ)
  
! TESTPARM_OBJ = utils/testparm.o \
!                $(PARAM_OBJ) $(LIB_NONSMBD_OBJ) $(POPT_LIB_OBJ) \
! 	       $(LIBSMB_ERR_OBJ)
  
! SMBTA_UTIL_OBJ = utils/smbta-util.o $(PARAM_OBJ) $(POPT_LIB_OBJ) \
! 	$(LIB_NONSMBD_OBJ) \
! 	$(LIBSMB_ERR_OBJ) $(FNAME_UTIL_OBJ)
! 
! TEST_LP_LOAD_OBJ = param/test_lp_load.o \
! 		   $(PARAM_OBJ) $(LIB_NONSMBD_OBJ) \
! 		   $(POPT_LIB_OBJ) $(LIBSAMBA_OBJ)
  
! PASSWD_UTIL_OBJ = utils/passwd_util.o
  
! SMBPASSWD_OBJ = utils/smbpasswd.o $(PASSWD_UTIL_OBJ) $(PASSCHANGE_OBJ) \
! 		$(PARAM_OBJ) $(LIBSMB_OBJ) $(PASSDB_OBJ) \
! 		$(GROUPDB_OBJ) $(LIB_NONSMBD_OBJ) $(KRBCLIENT_OBJ) \
! 		$(POPT_LIB_OBJ) $(SMBLDAP_OBJ) \
! 		$(LIBMSRPC_GEN_OBJ) $(LIBMSRPC_OBJ) \
! 		$(LIBCLI_SAMR_OBJ) \
! 		rpc_client/init_lsa.o
! 
! PDBEDIT_OBJ = utils/pdbedit.o $(PASSWD_UTIL_OBJ) $(PARAM_OBJ) $(PASSDB_OBJ) \
! 		$(LIBSAMBA_OBJ) $(LIBTSOCKET_OBJ) \
! 		$(LIB_NONSMBD_OBJ) $(GROUPDB_OBJ) \
! 		$(LIBCLI_LDAP_NDR_OBJ) \
! 		$(POPT_LIB_OBJ) $(SMBLDAP_OBJ) ../lib/util/asn1.o
  
! SMBGET_OBJ = utils/smbget.o $(POPT_LIB_OBJ) $(LIBSMBCLIENT_OBJ1)
  
  DISPLAY_SEC_OBJ= ../libcli/security/display_sec.o
  
! RPCCLIENT_OBJ1 = rpcclient/rpcclient.o rpcclient/cmd_lsarpc.o \
  	         rpcclient/cmd_samr.o rpcclient/cmd_spoolss.o \
  		 rpcclient/cmd_netlogon.o rpcclient/cmd_srvsvc.o \
  		 rpcclient/cmd_dfs.o rpcclient/cmd_epmapper.o \
--- 949,995 ----
  	       ../libcli/smb/smb2_create_blob.o \
  	       $(MANGLE_OBJ) @VFS_STATIC@
  
! SMBD_OBJ_BASE = $(SMBD_OBJ_SRV) \
! 		$(AUTH_OBJ) $(AVAHI_OBJ) $(BUILDOPT_OBJ) \
! 		$(LIBADS_SERVER_OBJ) $(LIBADS_PRINTER_OBJ) \
! 		$(LIBNET_OBJ) $(LIBNDR_XATTR_OBJ) \
! 		$(LOCKING_OBJ) $(NOTIFY_OBJ) $(OPLOCK_OBJ) \
! 		$(PRINTBACKEND_OBJ) $(PRINTING_OBJ) \
! 		$(REG_FULL_OBJ) $(REG_API_REGF_OBJ) $(RPC_SERVER_OBJ)
  
! SMBD_OBJ = $(SMBD_OBJ_BASE) $(SMBD_OBJ_MAIN)
  
! SMBGET_OBJ = utils/smbget.o
  
! SMBFILTER_OBJ = utils/smbfilter.o
! 
! SMBPASSWD_OBJ = utils/smbpasswd.o \
! 		$(PASSCHANGE_OBJ) $(PASSWD_UTIL_OBJ)
  
! SMBTA_UTIL_OBJ = utils/smbta-util.o
! 
! SMBTREE_OBJ = utils/smbtree.o
  
  STATUS_OBJ = utils/status.o utils/status_profile.o \
! 	     $(LOCKING_OBJ) $(PROFILE_OBJ)
  
! SWAT_OBJ = web/cgi.o web/diagnose.o web/startstop.o web/statuspage.o \
!            web/swat.o web/neg_lang.o \
! 	   $(LOCKING_OBJ) $(PASSCHANGE_OBJ) \
! 	   $(PLAINTEXT_AUTH_OBJ) $(PRINTING_OBJ) 
  
! TEST_LP_LOAD_OBJ = param/test_lp_load.o
  
! TESTPARM_OBJ = utils/testparm.o
  
! PASSWD_UTIL_OBJ = utils/passwd_util.o
  
! PDBEDIT_OBJ = utils/pdbedit.o \
! 	      $(PASSWD_UTIL_OBJ)
  
  DISPLAY_SEC_OBJ= ../libcli/security/display_sec.o
  
! RPCCLIENT_OBJ0 = rpcclient/rpcclient.o rpcclient/cmd_lsarpc.o \
  	         rpcclient/cmd_samr.o rpcclient/cmd_spoolss.o \
  		 rpcclient/cmd_netlogon.o rpcclient/cmd_srvsvc.o \
  		 rpcclient/cmd_dfs.o rpcclient/cmd_epmapper.o \
***************
*** 1045,1082 ****
  		 rpcclient/cmd_winreg.o \
  		 $(DISPLAY_SEC_OBJ)
  
! RPCCLIENT_OBJ = $(RPCCLIENT_OBJ1) \
!              $(PARAM_OBJ) $(LIBSMB_OBJ) $(LIB_NONSMBD_OBJ) \
!              $(PASSDB_OBJ) $(LIBMSRPC_GEN_OBJ) $(LIBMSRPC_OBJ) \
!              $(READLINE_OBJ) $(GROUPDB_OBJ) $(KRBCLIENT_OBJ) \
! 	     $(LIBADS_OBJ) $(POPT_LIB_OBJ) \
! 	     $(SMBLDAP_OBJ) $(DCUTIL_OBJ) \
! 	     $(LIBCLI_SPOOLSS_OBJ) \
! 	     $(LIBCLI_EVENTLOG_OBJ) \
! 	     $(LIBCLI_NTSVCS_OBJ) \
! 	     $(LIBCLI_DRSUAPI_OBJ) \
! 	     $(LIBCLI_DFS_OBJ) \
! 	     $(LIBCLI_ECHO_OBJ) \
! 	     $(LIBCLI_DSSETUP_OBJ) \
! 	     $(LIBCLI_WKSSVC_OBJ) \
! 	     $(LIBCLI_SRVSVC_OBJ) \
! 	     $(LIBCLI_LSA_OBJ) \
! 	     $(LIBCLI_SAMR_OBJ) \
! 	     $(LIBCLI_WINREG_OBJ) \
! 	     $(LIBCLI_NETLOGON_OBJ) \
! 	     $(RPC_CLIENT_SCHANNEL_OBJ) \
! 	     rpc_client/init_netlogon.o \
! 	     rpc_client/init_samr.o
  
  PAM_WINBIND_OBJ = ../nsswitch/pam_winbind.o $(WBCOMMON_OBJ) \
  		  $(LIBREPLACE_OBJ) @BUILD_INIPARSER@
  
! LIBSMBCLIENT_THREAD_OBJ = \
! 			libsmb/libsmb_thread_impl.o \
! 			libsmb/libsmb_thread_posix.o
  
! LIBSMBCLIENT_OBJ0 = \
! 		    libsmb/libsmb_cache.o \
  		    libsmb/libsmb_compat.o \
  		    libsmb/libsmb_context.o \
  		    libsmb/libsmb_dir.o \
--- 1000,1025 ----
  		 rpcclient/cmd_winreg.o \
  		 $(DISPLAY_SEC_OBJ)
  
! RPCCLIENT_OBJ = $(RPCCLIENT_OBJ0) \
! 		$(READLINE_OBJ) \
! 		$(LIBCLI_SPOOLSS_OBJ) \
! 		$(LIBCLI_EVENTLOG_OBJ) \
! 		$(LIBCLI_NTSVCS_OBJ) \
! 		$(LIBCLI_DRSUAPI_OBJ) \
! 		$(LIBCLI_DFS_OBJ) \
! 		$(LIBCLI_ECHO_OBJ) \
! 		$(LIBCLI_DSSETUP_OBJ) \
! 		$(LIBCLI_WKSSVC_OBJ) \
! 		$(LIBCLI_WINREG_OBJ) \
! 		$(RPC_CLIENT_SCHANNEL_OBJ)
  
  PAM_WINBIND_OBJ = ../nsswitch/pam_winbind.o $(WBCOMMON_OBJ) \
  		  $(LIBREPLACE_OBJ) @BUILD_INIPARSER@
  
! LIBSMBCLIENT_THREAD_OBJ = libsmb/libsmb_thread_impl.o \
! 			  libsmb/libsmb_thread_posix.o
  
! LIBSMBCLIENT_OBJ0 = libsmb/libsmb_cache.o \
  		    libsmb/libsmb_compat.o \
  		    libsmb/libsmb_context.o \
  		    libsmb/libsmb_dir.o \
***************
*** 1089,1124 ****
  		    libsmb/libsmb_xattr.o \
  		    libsmb/libsmb_setget.o
  
! LIBSMBCLIENT_OBJ1 = $(LIBSMBCLIENT_OBJ0) \
! 		    $(PARAM_OBJ) $(LIB_NONSMBD_OBJ) \
! 		    $(LIBSMB_OBJ) $(KRBCLIENT_OBJ) \
! 		    $(LIBMSRPC_OBJ) $(LIBMSRPC_GEN_OBJ) \
! 		    $(PASSDB_OBJ) $(SMBLDAP_OBJ) $(GROUPDB_OBJ) \
! 		    $(LIBCLI_SRVSVC_OBJ) \
! 		    $(LIBCLI_LSA_OBJ)
! 
! LIBSMBCLIENT_OBJ = $(LIBSMBCLIENT_OBJ1)
! 
! # This shared library is intended for linking with unit test programs
! # to test Samba internals.  It's called libbigballofmud.so to
! # discourage casual usage.
! 
! LIBBIGBALLOFMUD_MAJOR = 0
! 
! LIBBIGBALLOFMUD_OBJ = $(PARAM_OBJ) $(LIB_NONSMBD_OBJ) \
! 	$(LIBSMB_OBJ) $(LIBMSRPC_OBJ) $(LIBMSRPC_GEN_OBJ) $(PASSDB_OBJ) \
! 	$(GROUPDB_OBJ) $(KRBCLIENT_OBJ) $(SMBLDAP_OBJ)
! 
! CLIENT_OBJ1 = client/client.o client/clitar.o \
! 	      client/dnsbrowse.o \
! 	      $(LIBMSRPC_OBJ)
! 
! CLIENT_OBJ = $(CLIENT_OBJ1) $(PARAM_OBJ) $(LIBSMB_OBJ) \
! 	     $(LIB_NONSMBD_OBJ) $(KRBCLIENT_OBJ) $(LIBMSRPC_GEN_OBJ) \
!              $(READLINE_OBJ) $(POPT_LIB_OBJ) \
!              $(PASSDB_OBJ) $(SMBLDAP_OBJ) $(GROUPDB_OBJ) \
! 	     $(DISPLAY_SEC_OBJ) \
! 	     $(LIBCLI_SRVSVC_OBJ)
  
  LIBSMBCONF_OBJ = ../lib/smbconf/smbconf.o \
  		 ../lib/smbconf/smbconf_util.o \
--- 1032,1052 ----
  		    libsmb/libsmb_xattr.o \
  		    libsmb/libsmb_setget.o
  
! LIBSMBCLIENT_OBJ = $(LIBSMBCLIENT_OBJ0) \
! 		   rpc_client/init_netlogon.o \
! 		   rpc_client/init_samr.o \
! 		   $(DCUTIL_OBJ) $(FNAME_UTIL_OBJ) \
! 		   $(GROUPDB_OBJ) $(KRBCLIENT_OBJ) \
! 		   $(LIBCLI_LSA_OBJ) $(LIBCLI_NETLOGON_OBJ) \
! 		   $(LIBCLI_SAMR_OBJ) $(LIBCLI_SRVSVC_OBJ) \
! 		   $(LIBADS_OBJ) $(LIB_NONSMBD_OBJ) $(LIBSMB_OBJ) \
! 		   $(LIBMSRPC_OBJ) $(LIBMSRPC_GEN_OBJ) \
! 		   $(PARAM_OBJ) $(PASSDB_OBJ) \
! 		   $(POPT_OBJ) $(POPT_LIB_OBJ) \
! 		   $(REGFIO_OBJ) $(SMBLDAP_OBJ)
! 
! CLIENT_OBJ = client/client.o client/clitar.o client/dnsbrowse.o \
!              $(READLINE_OBJ) $(DISPLAY_SEC_OBJ)
  
  LIBSMBCONF_OBJ = ../lib/smbconf/smbconf.o \
  		 ../lib/smbconf/smbconf_util.o \
***************
*** 1126,1141 ****
  		 lib/smbconf/smbconf_reg.o \
  		 lib/smbconf/smbconf_init.o
  
! SMBCONFTORT_OBJ0 = lib/smbconf/testsuite.o
! 
! SMBCONFTORT_OBJ = $(SMBCONFTORT_OBJ0) \
! 		  $(LIB_NONSMBD_OBJ) \
! 		  $(PARAM_OBJ) \
! 		  $(LIBSMB_ERR_OBJ) \
! 		  $(POPT_LIB_OBJ)
  
  PTHREADPOOLTEST_OBJ = lib/pthreadpool/pthreadpool.o \
! 		lib/pthreadpool/tests.o
  
  LIBNET_OBJ = libnet/libnet_join.o \
  	     libnet/libnet_keytab.o \
--- 1054,1063 ----
  		 lib/smbconf/smbconf_reg.o \
  		 lib/smbconf/smbconf_init.o
  
! SMBCONFTORT_OBJ = lib/smbconf/testsuite.o
  
  PTHREADPOOLTEST_OBJ = lib/pthreadpool/pthreadpool.o \
! 		      lib/pthreadpool/tests.o
  
  LIBNET_OBJ = libnet/libnet_join.o \
  	     libnet/libnet_keytab.o \
***************
*** 1153,1159 ****
  		     libnet/libnet_samsync_keytab.o \
  		     ../libcli/samsync/decrypt.o
  
! NET_OBJ1 = utils/net.o utils/net_ads.o utils/net_help.o \
  	   utils/net_rap.o utils/net_rpc.o utils/net_rpc_samsync.o \
  	   utils/net_rpc_join.o utils/net_time.o utils/net_lookup.o \
  	   utils/net_cache.o utils/net_groupmap.o utils/net_idmap.o utils/net_idmap_check.o\
--- 1075,1081 ----
  		     libnet/libnet_samsync_keytab.o \
  		     ../libcli/samsync/decrypt.o
  
! NET_OBJ0 = utils/net.o utils/net_ads.o utils/net_help.o \
  	   utils/net_rap.o utils/net_rpc.o utils/net_rpc_samsync.o \
  	   utils/net_rpc_join.o utils/net_time.o utils/net_lookup.o \
  	   utils/net_cache.o utils/net_groupmap.o utils/net_idmap.o utils/net_idmap_check.o\
***************
*** 1179,1293 ****
  	   $(LIBCLI_DSSETUP_OBJ) \
  	   $(LIBCLI_SVCCTL_OBJ) \
  	   $(LIBCLI_WKSSVC_OBJ) \
- 	   $(LIBCLI_SRVSVC_OBJ) \
- 	   $(LIBCLI_LSA_OBJ) \
- 	   $(LIBCLI_SAMR_OBJ) \
- 	   $(LIBCLI_NETLOGON_OBJ) \
  	   $(RPC_CLIENT_SCHANNEL_OBJ) \
- 	   rpc_client/init_netlogon.o \
- 	   rpc_client/init_samr.o \
  	   registry/reg_parse.o registry/reg_format.o \
  	   registry/reg_parse_internal.o registry/reg_import.o \
  	   lib/cbuf.o lib/srprs.o printing/nt_printing_migrate.o
  
  # these are not processed by make proto
! NET_OBJ2 = utils/net_registry_util.o utils/net_help_common.o
  
! NET_OBJ = $(NET_OBJ1) \
! 	  $(NET_OBJ2) @FAKE_KASERVER_OBJ@\
! 	  $(PARAM_WITHOUT_REG_OBJ) $(LIBSMB_OBJ) \
! 	  $(PASSDB_OBJ) $(GROUPDB_OBJ) \
! 	  $(KRBCLIENT_OBJ) $(LIB_NONSMBD_OBJ) $(LIBADDNS_OBJ0) \
! 	  $(LIBMSRPC_OBJ) $(LIBMSRPC_GEN_OBJ) \
! 	  $(LIBADS_OBJ) $(LIBADS_SERVER_OBJ) $(LIBADS_PRINTER_OBJ) $(POPT_LIB_OBJ) \
! 	  $(SMBLDAP_OBJ) $(DCUTIL_OBJ) \
! 	  $(AFS_OBJ) $(AFS_SETTOKEN_OBJ) $(READLINE_OBJ) \
! 	  $(LIBGPO_OBJ) @BUILD_INIPARSER@ $(DISPLAY_SEC_OBJ) \
! 	  $(REG_SMBCONF_OBJ) \
  	  $(LIBNET_OBJ) $(LIBNET_DSSYNC_OBJ) $(LIBNET_SAMSYNC_OBJ) \
! 	  $(LIBSMBCONF_OBJ) \
! 	  $(REGFIO_OBJ) \
! 	  $(PRIVILEGES_BASIC_OBJ) \
! 	  $(LIB_EVENTLOG_OBJ)
! 
! CUPS_OBJ = client/smbspool.o $(PARAM_OBJ) $(LIBSMB_OBJ) \
! 	  $(LIB_NONSMBD_OBJ) $(KRBCLIENT_OBJ) $(POPT_LIB_OBJ) \
! 	  $(LIBNDR_GEN_OBJ0)
! 
! NMBLOOKUP_OBJ = utils/nmblookup.o $(PARAM_OBJ) $(LIBNMB_OBJ) \
!                $(LIB_NONSMBD_OBJ) $(POPT_LIB_OBJ) $(LIBSMB_ERR_OBJ)
! 
! SMBTORTURE_OBJ1 = torture/torture.o torture/nbio.o torture/scanner.o torture/utable.o \
! 		torture/denytest.o torture/mangle_test.o \
! 		torture/nbench.o \
! 		torture/test_async_echo.o \
! 		torture/test_smbsock_any_connect.o \
! 		torture/test_addrchange.o \
! 		torture/test_posix_append.o
! 
! SMBTORTURE_OBJ = $(SMBTORTURE_OBJ1) $(PARAM_OBJ) $(TLDAP_OBJ) \
! 	$(LIBSMB_OBJ) $(KRBCLIENT_OBJ) $(LIB_NONSMBD_OBJ) \
! 	@LIBWBCLIENT_STATIC@ \
!         torture/wbc_async.o \
!         ../nsswitch/wb_reqtrans.o \
! 	$(LIBMSRPC_OBJ) $(LIBMSRPC_GEN_OBJ) $(LIBCLI_ECHO_OBJ)
! 
! MASKTEST_OBJ = torture/masktest.o $(PARAM_OBJ) $(LIBSMB_OBJ) $(KRBCLIENT_OBJ) \
!                  $(LIB_NONSMBD_OBJ) \
! 		 $(LIBNDR_GEN_OBJ0)
! 
! MSGTEST_OBJ = torture/msgtest.o $(PARAM_OBJ) $(LIBSMB_ERR_OBJ) \
!                  $(LIB_NONSMBD_OBJ) \
! 		 $(LIBNDR_GEN_OBJ0)
! 
! LOCKTEST_OBJ = torture/locktest.o $(PARAM_OBJ) $(LOCKING_OBJ) $(KRBCLIENT_OBJ) \
!                $(LIBSMB_OBJ) $(LIB_NONSMBD_OBJ) \
!                $(LIBNDR_GEN_OBJ0) $(FNAME_UTIL_OBJ)
! 
! NSSTEST_OBJ = ../nsswitch/nsstest.o $(LIBSAMBAUTIL_OBJ)
! 
! PDBTEST_OBJ = torture/pdbtest.o $(PARAM_OBJ) $(LIBSMB_OBJ) $(KRBCLIENT_OBJ) \
! 		$(LIB_NONSMBD_OBJ) $(PASSDB_OBJ) $(GROUPDB_OBJ) \
! 		$(SMBLDAP_OBJ) $(POPT_LIB_OBJ) \
! 		$(LIBNDR_GEN_OBJ0)
  
! VFSTEST_OBJ = torture/cmd_vfs.o torture/vfstest.o $(SMBD_OBJ_BASE) $(READLINE_OBJ)
  
! SMBICONV_OBJ = $(PARAM_OBJ) torture/smbiconv.o $(LIB_NONSMBD_OBJ) $(POPT_LIB_OBJ) $(LIBSMB_ERR_OBJ)
  
  LOG2PCAP_OBJ = utils/log2pcaphex.o
  
! LOCKTEST2_OBJ = torture/locktest2.o $(PARAM_OBJ) $(LOCKING_OBJ) $(LIBSMB_OBJ) \
! 		$(KRBCLIENT_OBJ) $(LIB_NONSMBD_OBJ) \
! 		$(LIBNDR_GEN_OBJ0) $(FNAME_UTIL_OBJ)
! 
! SMBCACLS_OBJ = utils/smbcacls.o $(PARAM_OBJ) $(LIBSMB_OBJ) \
! 		$(KRBCLIENT_OBJ) $(LIB_NONSMBD_OBJ) \
! 		$(PASSDB_OBJ) $(GROUPDB_OBJ) $(LIBMSRPC_OBJ) $(LIBMSRPC_GEN_OBJ) \
! 		$(POPT_LIB_OBJ) $(SMBLDAP_OBJ) \
! 		$(LIBCLI_LSA_OBJ)
! 
! SMBCQUOTAS_OBJ = utils/smbcquotas.o $(LIBSMB_OBJ) $(KRBCLIENT_OBJ) \
! 		$(PARAM_OBJ) \
! 		$(LIB_NONSMBD_OBJ) \
! 		$(LIBMSRPC_OBJ) $(LIBMSRPC_GEN_OBJ) $(POPT_LIB_OBJ) \
! 		$(PASSDB_OBJ) $(SMBLDAP_OBJ) $(GROUPDB_OBJ) \
! 		$(LIBCLI_LSA_OBJ)
! 
! EVTLOGADM_OBJ0	= utils/eventlogadm.o
! 
! EVTLOGADM_OBJ	= $(EVTLOGADM_OBJ0) $(PARAM_OBJ) $(LIB_NONSMBD_OBJ) \
! 		$(LIBSMB_ERR_OBJ) $(LIB_EVENTLOG_OBJ) \
! 		librpc/gen_ndr/ndr_eventlog.o \
! 		librpc/gen_ndr/ndr_lsa.o
! 
! SHARESEC_OBJ0 = utils/sharesec.o
! SHARESEC_OBJ  = $(SHARESEC_OBJ0) $(PARAM_OBJ) $(LIB_NONSMBD_OBJ) \
! 		$(LIBSMB_ERR_OBJ) \
!                 $(POPT_LIB_OBJ)
  
! TALLOCTORT_OBJ = @tallocdir@/testsuite.o @tallocdir@/testsuite_main.o \
! 		$(PARAM_OBJ) $(LIB_NONSMBD_OBJ) $(LIBSMB_ERR_OBJ)
  
  REPLACETORT_OBJ = @libreplacedir@/test/testsuite.o \
  		@libreplacedir@/test/getifaddrs.o \
--- 1101,1159 ----
  	   $(LIBCLI_DSSETUP_OBJ) \
  	   $(LIBCLI_SVCCTL_OBJ) \
  	   $(LIBCLI_WKSSVC_OBJ) \
  	   $(RPC_CLIENT_SCHANNEL_OBJ) \
  	   registry/reg_parse.o registry/reg_format.o \
  	   registry/reg_parse_internal.o registry/reg_import.o \
  	   lib/cbuf.o lib/srprs.o printing/nt_printing_migrate.o
  
  # these are not processed by make proto
! NET_OBJ1 = utils/net_registry_util.o utils/net_help_common.o
  
! NET_OBJ = $(NET_OBJ0) $(NET_OBJ1) \
! 	  $(AFS_OBJ) $(AFS_SETTOKEN_OBJ) $(DISPLAY_SEC_OBJ) $(INIPARSER_OBJ) \
! 	  $(LIB_EVENTLOG_OBJ) $(LIBGPO_OBJ) \
  	  $(LIBNET_OBJ) $(LIBNET_DSSYNC_OBJ) $(LIBNET_SAMSYNC_OBJ) \
! 	  $(READLINE_OBJ)
! 
! CUPS_OBJ = client/smbspool.o
! 
! NMBLOOKUP_OBJ = utils/nmblookup.o
! 
! SMBTORTURE_OBJ = torture/torture.o torture/nbio.o torture/scanner.o \
! 		 torture/utable.o torture/denytest.o torture/mangle_test.o \
! 		 torture/nbench.o torture/test_async_echo.o \
! 		 torture/test_smbsock_any_connect.o torture/test_addrchange.o \
! 		 torture/test_posix_append.o torture/wbc_async.o \
! 		 ../nsswitch/wb_reqtrans.o \
! 		 $(LIBCLI_ECHO_OBJ) $(TLDAP_OBJ)
! 
! LOCKTEST_OBJ = torture/locktest.o \
!                $(LOCKING_OBJ)
! 
! LOCKTEST2_OBJ = torture/locktest2.o \
!                 $(LOCKING_OBJ)
! 
! MASKTEST_OBJ = torture/masktest.o
! 
! MSGTEST_OBJ = torture/msgtest.o
! 
! NSSTEST_OBJ = ../nsswitch/nsstest.o
! 
! PDBTEST_OBJ = torture/pdbtest.o
  
! VFSTEST_OBJ = torture/cmd_vfs.o torture/vfstest.o \
! 	      $(SMBD_OBJ_BASE) $(READLINE_OBJ)
  
! SMBICONV_OBJ = torture/smbiconv.o
  
  LOG2PCAP_OBJ = utils/log2pcaphex.o
  
! EVTLOGADM_OBJ = utils/eventlogadm.o \
! 		$(LIB_EVENTLOG_OBJ)
  
! SHARESEC_OBJ = utils/sharesec.o
! 
! TALLOCTORT_OBJ = @tallocdir@/testsuite.o @tallocdir@/testsuite_main.o
  
  REPLACETORT_OBJ = @libreplacedir@/test/testsuite.o \
  		@libreplacedir@/test/getifaddrs.o \
***************
*** 1298,1351 ****
  
  DEBUG2HTML_OBJ = utils/debug2html.o utils/debugparse.o
  
! SMBFILTER_OBJ = utils/smbfilter.o $(PARAM_OBJ) $(LIBSMB_OBJ) \
!                  $(LIB_NONSMBD_OBJ) $(KRBCLIENT_OBJ) \
! 		 $(LIBNDR_GEN_OBJ0)
! 
! WINBIND_WINS_NSS_OBJ = ../nsswitch/wins.o $(PARAM_OBJ) \
! 	$(LIB_NONSMBD_OBJ) $(LIBSMB_ERR_OBJ) $(LIBNMB_OBJ)
! 
! PAM_SMBPASS_OBJ_0 = pam_smbpass/pam_smb_auth.o pam_smbpass/pam_smb_passwd.o \
! 		pam_smbpass/pam_smb_acct.o pam_smbpass/support.o ../lib/util/asn1.o
! PAM_SMBPASS_OBJ = $(PAM_SMBPASS_OBJ_0) $(PARAM_OBJ) $(LIB_NONSMBD_OBJ) $(PASSDB_OBJ) $(GROUPDB_OBJ) \
! 		$(SMBLDAP_OBJ) $(LIBSAMBA_OBJ) \
! 		$(LIBTSOCKET_OBJ) $(PAM_ERRORS_OBJ)
  
  IDMAP_RW_OBJ = winbindd/idmap_rw.o
  
! IDMAP_OBJ     = winbindd/idmap.o winbindd/idmap_util.o $(IDMAP_RW_OBJ) @IDMAP_STATIC@
  
  NSS_INFO_OBJ = winbindd/nss_info.o @NSS_INFO_STATIC@
  
! IDMAP_HASH_OBJ = \
! 		winbindd/idmap_hash/idmap_hash.o \
! 		winbindd/idmap_hash/mapfile.o
! 
! IDMAP_ADEX_OBJ = \
! 		winbindd/idmap_adex/idmap_adex.o \
! 		winbindd/idmap_adex/cell_util.o \
! 		winbindd/idmap_adex/likewise_cell.o \
! 		winbindd/idmap_adex/provider_unified.o \
! 		winbindd/idmap_adex/gc_util.o \
! 		winbindd/idmap_adex/domain_util.o
  
! WINBINDD_OBJ1 = \
! 		winbindd/winbindd.o       \
  		winbindd/winbindd_group.o \
! 		winbindd/winbindd_util.o  \
  		winbindd/winbindd_cache.o \
! 		winbindd/winbindd_pam.o   \
! 		winbindd/winbindd_misc.o  \
! 		winbindd/winbindd_cm.o    \
! 		winbindd/winbindd_wins.o  \
  		winbindd/winbindd_msrpc.o \
! 		winbindd/winbindd_rpc.o   \
  		winbindd/winbindd_reconnect.o \
! 		winbindd/winbindd_ads.o   \
  		winbindd/winbindd_samr.o \
! 		winbindd/winbindd_dual.o  \
! 		winbindd/winbindd_dual_ndr.o  \
! 		winbindd/winbindd_dual_srv.o  \
  		librpc/gen_ndr/ndr_wbint_c.o \
  		librpc/gen_ndr/srv_wbint.o \
  		librpc/gen_ndr/ndr_wbint.o \
--- 1164,1209 ----
  
  DEBUG2HTML_OBJ = utils/debug2html.o utils/debugparse.o
  
! WINBIND_WINS_NSS_OBJ = ../nsswitch/wins.o \
! 		       $(LIBSMB_ERR_OBJ) $(LIBNMB_OBJ)
! 
! PAM_SMBPASS_OBJ = pam_smbpass/pam_smb_auth.o pam_smbpass/pam_smb_passwd.o \
! 		  pam_smbpass/pam_smb_acct.o pam_smbpass/support.o \
! 		  $(LIBTSOCKET_OBJ) $(PAM_ERRORS_OBJ) $(SMBLDAP_OBJ)
  
  IDMAP_RW_OBJ = winbindd/idmap_rw.o
  
! IDMAP_OBJ = winbindd/idmap.o winbindd/idmap_util.o \
! 	    $(IDMAP_RW_OBJ) @IDMAP_STATIC@
  
  NSS_INFO_OBJ = winbindd/nss_info.o @NSS_INFO_STATIC@
  
! IDMAP_HASH_OBJ = winbindd/idmap_hash/mapfile.o \
! 		 winbindd/idmap_hash/idmap_hash.o \
! 
! IDMAP_ADEX_OBJ = winbindd/idmap_adex/idmap_adex.o \
! 		 winbindd/idmap_adex/cell_util.o \
! 		 winbindd/idmap_adex/likewise_cell.o \
! 		 winbindd/idmap_adex/provider_unified.o \
! 		 winbindd/idmap_adex/gc_util.o \
! 		 winbindd/idmap_adex/domain_util.o
  
! WINBINDD_OBJ0 = winbindd/winbindd.o \
  		winbindd/winbindd_group.o \
! 		winbindd/winbindd_util.o \
  		winbindd/winbindd_cache.o \
! 		winbindd/winbindd_pam.o \
! 		winbindd/winbindd_misc.o \
! 		winbindd/winbindd_cm.o \
! 		winbindd/winbindd_wins.o \
  		winbindd/winbindd_msrpc.o \
! 		winbindd/winbindd_rpc.o \
  		winbindd/winbindd_reconnect.o \
! 		winbindd/winbindd_ads.o \
  		winbindd/winbindd_samr.o \
! 		winbindd/winbindd_dual.o \
! 		winbindd/winbindd_dual_ndr.o \
! 		winbindd/winbindd_dual_srv.o \
  		librpc/gen_ndr/ndr_wbint_c.o \
  		librpc/gen_ndr/srv_wbint.o \
  		librpc/gen_ndr/ndr_wbint.o \
***************
*** 1429,1529 ****
  		auth/pampass.o \
  		../nsswitch/wb_reqtrans.o
  
! WINBINDD_OBJ = \
! 		$(WINBINDD_OBJ1) $(PASSDB_OBJ) $(GROUPDB_OBJ) \
! 		$(PARAM_OBJ) $(LIB_NONSMBD_OBJ) \
! 		$(LIBSMB_OBJ) $(LIBMSRPC_OBJ) $(LIBMSRPC_GEN_OBJ) \
! 		$(PROFILE_OBJ) $(SLCACHE_OBJ) $(SMBLDAP_OBJ) \
! 		$(LIBADS_OBJ) $(KRBCLIENT_OBJ) $(POPT_LIB_OBJ) \
! 		$(DCUTIL_OBJ) $(IDMAP_OBJ) $(NSS_INFO_OBJ) \
! 		$(RPC_NCACN_NP) $(RPC_SAMR_OBJ) $(RPC_LSARPC_OBJ) \
! 		$(NPA_TSTREAM_OBJ) \
! 		$(AFS_OBJ) $(AFS_SETTOKEN_OBJ) \
! 		$(LIBADS_SERVER_OBJ) \
! 		$(TDB_VALIDATE_OBJ) \
! 		$(LIBCLI_DSSETUP_OBJ) \
! 		$(LIBCLI_LSA_OBJ) \
! 		$(LIBCLI_SAMR_OBJ) \
! 		$(LIBCLI_NETLOGON_OBJ) \
! 		rpc_client/init_netlogon.o \
! 		rpc_client/init_samr.o \
! 		$(PAM_ERRORS_OBJ)
! 
! WBINFO_OBJ = ../nsswitch/wbinfo.o $(LIBSAMBA_OBJ) $(PARAM_OBJ) $(LIB_NONSMBD_OBJ) \
! 		$(POPT_LIB_OBJ) $(AFS_SETTOKEN_OBJ) \
! 		lib/winbind_util.o $(WBCOMMON_OBJ)
  
  WINBIND_NSS_OBJ = $(WBCOMMON_OBJ) $(LIBREPLACE_OBJ) @WINBIND_NSS_EXTRA_OBJS@
  
  WINBIND_KRB5_LOCATOR_OBJ1 = ../nsswitch/winbind_krb5_locator.o
  WINBIND_KRB5_LOCATOR_OBJ = $(WINBIND_KRB5_LOCATOR_OBJ1) $(LIBREPLACE_OBJ)
  
! POPT_OBJ=../lib/popt/findme.o ../lib/popt/popt.o ../lib/popt/poptconfig.o \
!           ../lib/popt/popthelp.o ../lib/popt/poptparse.o
! 
! INIPARSER_OBJ = ../lib/iniparser_build/iniparser.o ../lib/iniparser_build/dictionary.o \
  		../lib/iniparser_build/strlib.o
  
! TDBBACKUP_OBJ = @tdbdir@/tools/tdbbackup.o $(LIBREPLACE_OBJ) \
! 	$(SOCKET_WRAPPER_OBJ)
! 
! TDBTOOL_OBJ = @tdbdir@/tools/tdbtool.o $(LIBREPLACE_OBJ) \
! 	$(SOCKET_WRAPPER_OBJ)
  
! TDBDUMP_OBJ = @tdbdir@/tools/tdbdump.o $(LIBREPLACE_OBJ) \
! 	$(SOCKET_WRAPPER_OBJ)
  
! TDBRESTORE_OBJ = @tdbdir@/tools/tdbrestore.o $(LIBREPLACE_OBJ) \
! 	$(SOCKET_WRAPPER_OBJ)
  
! TDBTORTURE_OBJ = @tdbdir@/tools/tdbtorture.o $(LIBREPLACE_OBJ) \
! 	$(SOCKET_WRAPPER_OBJ)
  
  
! NTLM_AUTH_OBJ1 = utils/ntlm_auth.o utils/ntlm_auth_diagnostics.o
  
! NTLM_AUTH_OBJ = ${NTLM_AUTH_OBJ1} $(LIBSAMBA_OBJ) $(POPT_LIB_OBJ) \
! 		../lib/util/asn1.o ../libcli/auth/spnego_parse.o libsmb/clikrb5.o libads/kerberos.o \
  		libsmb/samlogon_cache.o \
! 		$(LIBADS_SERVER_OBJ) \
! 		$(PASSDB_OBJ) $(LIBTSOCKET_OBJ) $(GROUPDB_OBJ) \
! 		$(SMBLDAP_OBJ) $(LIBNMB_OBJ) \
! 		$(WBCOMMON_OBJ) \
! 		$(LIBCLI_LDAP_NDR_OBJ) \
! 		$(LIBNDR_GEN_OBJ0) $(LIBNDR_NETLOGON_OBJ) @BUILD_INIPARSER@
! 
  
  VLP_OBJ = printing/tests/vlp.o \
  	  ../lib/util/util_tdb.o \
  	  $(LIBSAMBAUTIL_OBJ) \
  	  param/util.o
  
! RPC_OPEN_TCP_OBJ = torture/rpc_open_tcp.o \
! 		   $(LIBSMB_OBJ) \
! 		   $(PARAM_OBJ) \
! 		   $(PASSDB_OBJ) \
! 		   $(SMBLDAP_OBJ) $(GROUPDB_OBJ) \
! 		   $(LIB_NONSMBD_OBJ) \
! 		   $(KRBCLIENT_OBJ) \
! 		   $(LIBMSRPC_OBJ) \
! 		   $(LIBMSRPC_GEN_OBJ)
! 
! DBWRAP_TOOL_OBJ = utils/dbwrap_tool.o \
! 		  $(PARAM_OBJ) \
! 		  $(LIB_NONSMBD_OBJ) \
! 		  $(LIBSMB_ERR_OBJ)
! 
! DBWRAP_TORTURE_OBJ = utils/dbwrap_torture.o \
! 		     $(PARAM_OBJ) \
! 		     $(LIB_NONSMBD_OBJ) \
! 		     $(LIBSMB_ERR_OBJ) \
! 		     $(POPT_LIB_OBJ)
! 
! SPLIT_TOKENS_OBJ = utils/split_tokens.o \
!                    $(PARAM_OBJ) \
! 		   $(LIB_NONSMBD_OBJ) \
! 		   $(POPT_LIB_OBJ) \
! 	           $(LIBSAMBA_OBJ)
  
  ######################################################################
  # now the rules...
--- 1287,1345 ----
  		auth/pampass.o \
  		../nsswitch/wb_reqtrans.o
  
! WINBINDD_OBJ = $(WINBINDD_OBJ0) \
! 	       $(AFS_OBJ) $(AFS_SETTOKEN_OBJ) $(IDMAP_OBJ) \
! 	       $(LIBADS_SERVER_OBJ) $(LIBCLI_DSSETUP_OBJ) \
! 	       $(NPA_TSTREAM_OBJ) $(NSS_INFO_OBJ) $(PAM_ERRORS_OBJ) \
! 	       $(RPC_NCACN_NP) $(RPC_SAMR_OBJ) $(RPC_LSARPC_OBJ) \
! 	       $(SLCACHE_OBJ) $(TDB_VALIDATE_OBJ)
! 
! WBINFO_OBJ = ../nsswitch/wbinfo.o
  
  WINBIND_NSS_OBJ = $(WBCOMMON_OBJ) $(LIBREPLACE_OBJ) @WINBIND_NSS_EXTRA_OBJS@
  
  WINBIND_KRB5_LOCATOR_OBJ1 = ../nsswitch/winbind_krb5_locator.o
  WINBIND_KRB5_LOCATOR_OBJ = $(WINBIND_KRB5_LOCATOR_OBJ1) $(LIBREPLACE_OBJ)
  
! INIPARSER_OBJ = ../lib/iniparser_build/dictionary.o \
! 		../lib/iniparser_build/iniparser.o \
  		../lib/iniparser_build/strlib.o
  
! POPT_OBJ = ../lib/popt/findme.o ../lib/popt/popt.o ../lib/popt/poptconfig.o \
!            ../lib/popt/popthelp.o ../lib/popt/poptparse.o
  
! TDBBACKUP_OBJ = @tdbdir@/tools/tdbbackup.o \
! 		$(LIBREPLACE_OBJ) $(SOCKET_WRAPPER_OBJ)
  
! TDBTOOL_OBJ = @tdbdir@/tools/tdbtool.o \
! 	      $(LIBREPLACE_OBJ) $(SOCKET_WRAPPER_OBJ)
  
! TDBDUMP_OBJ = @tdbdir@/tools/tdbdump.o \
! 	      $(LIBREPLACE_OBJ) $(SOCKET_WRAPPER_OBJ)
  
+ TDBRESTORE_OBJ = @tdbdir@/tools/tdbrestore.o \
+ 		 $(LIBREPLACE_OBJ) $(SOCKET_WRAPPER_OBJ)
  
! TDBTORTURE_OBJ = @tdbdir@/tools/tdbtorture.o \
! 		 $(LIBREPLACE_OBJ) $(SOCKET_WRAPPER_OBJ)
  
! NTLM_AUTH_OBJ = utils/ntlm_auth.o utils/ntlm_auth_diagnostics.o \
  		libsmb/samlogon_cache.o \
!                 $(INIPARSER_OBJ) \
! 		$(LIBADS_SERVER_OBJ) $(LIBCLI_LDAP_NDR_OBJ) $(LIBTSOCKET_OBJ)
  
  VLP_OBJ = printing/tests/vlp.o \
  	  ../lib/util/util_tdb.o \
  	  $(LIBSAMBAUTIL_OBJ) \
  	  param/util.o
  
! RPC_OPEN_TCP_OBJ = torture/rpc_open_tcp.o
! 
! DBWRAP_TOOL_OBJ = utils/dbwrap_tool.o
! 
! DBWRAP_TORTURE_OBJ = utils/dbwrap_torture.o
! 
! SPLIT_TOKENS_OBJ = utils/split_tokens.o
  
  ######################################################################
  # now the rules...
***************
*** 1774,2000 ****
  	  dir=bin $(MAKEDIR); fi
  	@: >> $@ || : > $@ # what a fancy emoticon!
  
! bin/smbd@EXEEXT@: $(BINARY_PREREQS) $(SMBD_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBWBCLIENT) @BUILD_POPT@
  	@echo Linking $@
! 	@$(CC) -o $@ $(SMBD_OBJ) $(LDFLAGS) $(LDAP_LIBS) @SMBD_FAM_LIBS@ \
! 		$(KRB5LIBS) $(DYNEXP) $(PRINT_LIBS) $(AUTH_LIBS) \
! 		$(ACL_LIBS) $(PASSDB_LIBS) $(LIBS) $(DNSSD_LIBS) $(AVAHI_LIBS) \
! 		$(POPT_LIBS) @SMBD_LIBS@ $(LIBTALLOC_LIBS) $(LIBTDB_LIBS) \
! 		$(LIBWBCLIENT_LIBS) $(ZLIB_LIBS)
  
! bin/nmbd@EXEEXT@: $(BINARY_PREREQS) $(NMBD_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(NMBD_OBJ) $(LDFLAGS) $(DYNEXP) $(LIBS) \
! 		@LIBWBCLIENT_STATIC@ $(LIBWBCLIENT_LIBS) \
  		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(KRB5LIBS) $(LDAP_LIBS) $(ZLIB_LIBS)
  
! bin/swat@EXEEXT@: $(BINARY_PREREQS) $(SWAT_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(SWAT_OBJ) $(LDFLAGS) $(DYNEXP) $(PRINT_LIBS) \
! 	  $(AUTH_LIBS) $(LIBS) $(PASSDB_LIBS) $(POPT_LIBS) $(KRB5LIBS) \
! 	  $(LDAP_LIBS) $(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(LIBWBCLIENT_LIBS) $(ZLIB_LIBS)
  
! bin/rpcclient@EXEEXT@: $(BINARY_PREREQS) $(RPCCLIENT_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(RPCCLIENT_OBJ) \
! 		$(DYNEXP) $(TERMLDFLAGS) $(TERMLIBS) $(LIBS) $(POPT_LIBS) \
! 		$(KRB5LIBS) $(LDAP_LIBS) $(LIBTALLOC_LIBS) $(LIBTDB_LIBS) \
! 		$(LIBWBCLIENT_LIBS) $(ZLIB_LIBS) $(PASSDB_LIBS)
  
! bin/smbclient@EXEEXT@: $(BINARY_PREREQS) $(CLIENT_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(CLIENT_OBJ) $(LDFLAGS) $(DYNEXP) \
! 		$(TERMLDFLAGS) $(TERMLIBS) $(LIBS) $(POPT_LIBS) \
! 		$(KRB5LIBS) $(LDAP_LIBS) $(NSCD_LIBS) $(DNSSD_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(LIBWBCLIENT_LIBS) $(ZLIB_LIBS)
  
! bin/net@EXEEXT@: $(BINARY_PREREQS) $(NET_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB) $(LIBWBCLIENT) $(LIBNETAPI)
  	@echo Linking $@
! 	@$(CC) -o $@ $(NET_OBJ) $(DYNEXP) $(LDFLAGS) $(LIBS) \
! 		$(POPT_LIBS) $(KRB5LIBS) $(UUID_LIBS) $(LDAP_LIBS) \
! 		$(PASSDB_LIBS) $(TERMLDFLAGS) $(TERMLIBS) $(NSCD_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(LIBWBCLIENT_LIBS) $(LIBNETAPI_LIBS) \
! 		$(ZLIB_LIBS)
  
! bin/profiles@EXEEXT@: $(BINARY_PREREQS) $(PROFILES_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB)
  	@echo Linking $@
! 	@$(CC) -o $@ $(PROFILES_OBJ) $(DYNEXP) $(LDFLAGS) $(LIBS) \
! 		$(LDAP_LIBS) $(POPT_LIBS) $(LIBTALLOC_LIBS) $(LIBTDB_LIBS)
  
! bin/smbspool@EXEEXT@: $(BINARY_PREREQS) $(CUPS_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(CUPS_OBJ) $(DYNEXP) $(LDFLAGS) $(LIBS) \
! 		@LIBWBCLIENT_STATIC@ $(LIBWBCLIENT_LIBS) \
! 		$(KRB5LIBS) $(LDAP_LIBS) $(POPT_LIBS) $(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(ZLIB_LIBS)
  
! bin/testparm@EXEEXT@: $(BINARY_PREREQS) $(TESTPARM_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB)
  	@echo Linking $@
! 	@$(CC) -o $@ $(TESTPARM_OBJ) $(LDFLAGS) $(DYNEXP) $(LIBS) \
! 		$(LDAP_LIBS) $(POPT_LIBS) $(LIBTALLOC_LIBS) $(LIBTDB_LIBS)
  
! bin/smbta-util@EXEEXT@: $(BINARY_PREREQS) $(SMBTA_UTIL_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB)
  	@echo Linking $@
! 	@$(CC) -o $@ $(SMBTA_UTIL_OBJ) $(LDFLAGS) $(DYNEXP) $(LIBS) \
! 		$(LDAP_LIBS) $(POPT_LIBS) $(LIBTALLOC_LIBS) $(LIBTDB_LIBS)
  
! bin/smbstatus@EXEEXT@: $(BINARY_PREREQS) $(STATUS_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB)
  	@echo Linking $@
! 	@$(CC) -o $@ $(STATUS_OBJ) $(LDFLAGS) $(DYNEXP) $(LIBS) \
! 		$(LDAP_LIBS) $(POPT_LIBS) $(LIBTALLOC_LIBS) $(LIBTDB_LIBS)
  
! bin/smbcontrol@EXEEXT@: $(BINARY_PREREQS) $(SMBCONTROL_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB)
  	@echo Linking $@
! 	@$(CC) -DUSING_SMBCONTROL -o $@ \
! 		$(SMBCONTROL_OBJ) $(DYNEXP) $(LDFLAGS) \
! 		$(LIBS) $(LDAP_LIBS) @LIBUNWIND_PTRACE@ $(POPT_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS)
  
! bin/smbtree@EXEEXT@: $(BINARY_PREREQS) $(SMBTREE_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(SMBTREE_OBJ) $(LDFLAGS) $(DYNEXP) \
! 		$(LIBS) $(POPT_LIBS) $(KRB5LIBS) $(LDAP_LIBS) $(NSCD_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(LIBWBCLIENT_LIBS) $(ZLIB_LIBS)
  
! bin/smbpasswd@EXEEXT@: $(BINARY_PREREQS) $(SMBPASSWD_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(SMBPASSWD_OBJ) $(LDFLAGS) $(PASSDB_LIBS) \
! 		$(DYNEXP) $(LIBS) $(POPT_LIBS) $(KRB5LIBS) $(LDAP_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(LIBWBCLIENT_LIBS) $(ZLIB_LIBS)
  
! bin/pdbedit@EXEEXT@: $(BINARY_PREREQS) $(PDBEDIT_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(PDBEDIT_OBJ) $(LDFLAGS) $(DYNEXP) $(LIBS) \
! 		$(POPT_LIBS) $(PASSDB_LIBS) $(LDAP_LIBS) $(LIBTALLOC_LIBS) \
! 		$(LIBTDB_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/smbget@EXEEXT@: $(BINARY_PREREQS) $(SMBGET_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(SMBGET_OBJ) $(LDFLAGS) $(DYNEXP) $(LIBS) \
! 		$(POPT_LIBS)  $(KRB5LIBS) $(LDAP_LIBS) $(NSCD_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(LIBWBCLIENT_LIBS) $(ZLIB_LIBS)
  
! bin/nmblookup@EXEEXT@: $(BINARY_PREREQS) $(NMBLOOKUP_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB)
  	@echo Linking $@
! 	@$(CC) -o $@ $(NMBLOOKUP_OBJ) $(LDFLAGS) $(DYNEXP) $(LIBS) \
! 		$(POPT_LIBS) $(LDAP_LIBS) $(LIBTALLOC_LIBS) $(LIBTDB_LIBS)
  
! bin/smbtorture@EXEEXT@: $(BINARY_PREREQS) $(SMBTORTURE_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(SMBTORTURE_OBJ) $(LDFLAGS) $(DYNEXP) \
! 		$(LIBS) $(KRB5LIBS) $(LDAP_LIBS) $(POPT_LIBS) $(LIBTALLOC_LIBS) \
! 		$(LIBTDB_LIBS) $(ZLIB_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/talloctort@EXEEXT@: $(BINARY_PREREQS) $(TALLOCTORT_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB)
  	@echo Linking $@
! 	@$(CC) -o $@ $(TALLOCTORT_OBJ) $(LDFLAGS) \
! 		$(DYNEXP) $(LIBS) $(LDAP_LIBS) $(POPT_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS)
  
! bin/replacetort@EXEEXT@: $(REPLACETORT_OBJ) @BUILD_POPT@ $(LIBTALLOC)
  	@echo Linking $@
! 	@$(CC) -o $@ $(REPLACETORT_OBJ) $(LDFLAGS) \
! 		$(DYNEXP) $(LIBS) $(POPT_LIBS) $(LIBTALLOC_LIBS)
  
! bin/smbconftort@EXEEXT@: $(SMBCONFTORT_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB)
  	@echo Linking $@
! 	@$(CC) -o $@ $(SMBCONFTORT_OBJ) $(LDFLAGS) \
! 		$(DYNEXP) $(LIBS) $(LDAP_LIBS) $(POPT_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS)
  
! bin/pthreadpooltest@EXEEXT@: $(PTHREADPOOLTEST_OBJ)
  	@echo Linking $@
! 	@$(CC) -o $@ $(PTHREADPOOLTEST_OBJ) $(LDFLAGS) -lpthread
  
! bin/masktest@EXEEXT@: $(BINARY_PREREQS) $(MASKTEST_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(MASKTEST_OBJ) $(LDFLAGS) $(DYNEXP) \
! 		$(LIBS) $(KRB5LIBS) $(LDAP_LIBS) $(POPT_LIBS) \
! 		@LIBWBCLIENT_STATIC@ $(LIBWBCLIENT_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(ZLIB_LIBS)
  
! bin/msgtest@EXEEXT@: $(BINARY_PREREQS) $(MSGTEST_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB)
  	@echo Linking $@
! 	@$(CC) -o $@ $(MSGTEST_OBJ) $(LDFLAGS) $(DYNEXP) \
! 		$(LIBS) $(KRB5LIBS) $(LDAP_LIBS) $(POPT_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(ZLIB_LIBS)
  
! bin/smbcacls@EXEEXT@: $(BINARY_PREREQS) $(SMBCACLS_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(SMBCACLS_OBJ) $(DYNEXP) $(LDFLAGS) \
! 		$(LIBS) $(POPT_LIBS) $(KRB5LIBS) $(LDAP_LIBS) $(NSCD_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(LIBWBCLIENT_LIBS) $(ZLIB_LIBS)
  
! bin/smbcquotas@EXEEXT@: $(BINARY_PREREQS) $(SMBCQUOTAS_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(SMBCQUOTAS_OBJ) $(DYNEXP) $(LDFLAGS) \
! 		$(LIBS) $(POPT_LIBS) $(KRB5LIBS) $(LDAP_LIBS) $(NSCD_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(LIBWBCLIENT_LIBS) $(ZLIB_LIBS)
  
! bin/eventlogadm@EXEEXT@: $(BINARY_PREREQS) $(EVTLOGADM_OBJ) $(LIBTALLOC) $(LIBTDB)
  	@echo Linking $@
! 	@$(CC) -o $@ $(EVTLOGADM_OBJ) $(DYNEXP) $(LDFLAGS) \
! 		$(LIBS) $(LDAP_LIBS) $(LIBTALLOC_LIBS) $(LIBTDB_LIBS)
  
! bin/sharesec@EXEEXT@: $(BINARY_PREREQS) $(SHARESEC_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB)
  	@echo Linking $@
! 	@$(CC) -o $@ $(SHARESEC_OBJ) $(DYNEXP) $(LDFLAGS) \
! 		$(LIBS) $(LDAP_LIBS) $(POPT_LIBS) $(LIBTALLOC_LIBS) $(LIBTDB_LIBS)
  
! bin/locktest@EXEEXT@: $(BINARY_PREREQS) $(LOCKTEST_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LOCKTEST_OBJ) $(LDFLAGS) $(DYNEXP) \
! 		$(LIBS) $(KRB5LIBS) $(LDAP_LIBS) $(POPT_LIBS) \
! 		@LIBWBCLIENT_STATIC@ $(LIBWBCLIENT_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(ZLIB_LIBS)
  
  bin/nsstest@EXEEXT@: $(BINARY_PREREQS) $(NSSTEST_OBJ)
  	@echo Linking $@
! 	@$(CC) -o $@ $(NSSTEST_OBJ) $(LDFLAGS) $(DYNEXP) $(LIBS)
  
! bin/pdbtest@EXEEXT@: $(BINARY_PREREQS) $(PDBTEST_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(PDBTEST_OBJ) $(LDFLAGS) $(DYNEXP) \
! 		$(LIBS) $(KRB5LIBS) $(LDAP_LIBS) $(PASSDB_LIBS) \
! 		$(POPT_LIBS) $(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(LIBWBCLIENT_LIBS) $(ZLIB_LIBS)
  
! bin/vfstest@EXEEXT@: $(BINARY_PREREQS) $(VFSTEST_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(VFSTEST_OBJ) $(LDFLAGS) $(TERMLDFLAGS) $(AVAHI_LIBS) \
! 		$(TERMLIBS) $(DYNEXP) $(PRINT_LIBS) $(AUTH_LIBS) $(DNSSD_LIBS) \
! 		$(ACL_LIBS) $(LIBS) $(POPT_LIBS) $(KRB5LIBS) $(LDAP_LIBS) \
! 		@SMBD_LIBS@ $(NSCD_LIBS) $(LIBTALLOC_LIBS) $(LIBTDB_LIBS) \
! 		$(LIBWBCLIENT_LIBS) $(ZLIB_LIBS)
  
! bin/smbiconv@EXEEXT@: $(BINARY_PREREQS) $(SMBICONV_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB)
  	@echo Linking $@
! 	@$(CC) -o $@ $(SMBICONV_OBJ) $(LDFLAGS) $(TERMLDFLAGS) \
! 		$(TERMLIBS) $(DYNEXP) $(LIBS) $(LDAP_LIBS) $(POPT_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS)
  
! bin/log2pcap@EXEEXT@: $(BINARY_PREREQS) $(LOG2PCAP_OBJ) @BUILD_POPT@ $(LIBTALLOC)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LOG2PCAP_OBJ) $(LDFLAGS) $(DYNEXP) \
! 		$(POPT_LIBS) $(LIBS) $(LIBTALLOC_LIBS)
  
! bin/locktest2@EXEEXT@: $(BINARY_PREREQS) $(LOCKTEST2_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LOCKTEST2_OBJ) $(LDFLAGS) $(DYNEXP) \
! 		@LIBWBCLIENT_STATIC@ $(LIBWBCLIENT_LIBS) \
! 		$(LIBS) $(KRB5LIBS) $(LDAP_LIBS) $(POPT_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(ZLIB_LIBS)
  
! bin/debug2html@EXEEXT@: $(BINARY_PREREQS) $(DEBUG2HTML_OBJ) @BUILD_POPT@ $(LIBTALLOC)
  	@echo Linking $@
! 	@$(CC) -o $@ $(DEBUG2HTML_OBJ) $(LDFLAGS) $(DYNEXP) \
! 		$(LIBS) $(LIBTALLOC_LIBS)
  
! bin/smbfilter@EXEEXT@: $(BINARY_PREREQS) $(SMBFILTER_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(SMBFILTER_OBJ) $(LDFLAGS) $(DYNEXP) $(LIBS) \
! 		$(KRB5LIBS) $(LDAP_LIBS) $(POPT_LIBS) \
! 		@LIBWBCLIENT_STATIC@ $(LIBWBCLIENT_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(ZLIB_LIBS)
  
  bin/versiontest: $(BINARY_PREREQS) lib/version_test.o $(VERSION_OBJ)
  	@echo Linking $@
--- 1590,1858 ----
  	  dir=bin $(MAKEDIR); fi
  	@: >> $@ || : > $@ # what a fancy emoticon!
  
! bin/smbd@EXEEXT@: $(BINARY_PREREQS) $(SMBD_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(SMBD_OBJ) $(DYNEXP) $(LIBS) \
! 		$(KRB5LIBS) $(LDAP_LIBS) $(ZLIB_LIBS) \
! 		$(AUTH_LIBS) $(ACL_LIBS) $(AVAHI_LIBS) $(DNSSD_LIBS) \
! 		$(PASSDB_LIBS) $(PRINT_LIBS) \
! 		@SMBD_FAM_LIBS@ @SMBD_LIBS@ \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/nmbd@EXEEXT@: $(BINARY_PREREQS) $(NMBD_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(NMBD_OBJ) $(DYNEXP) $(LIBS) \
! 		$(KRB5LIBS) $(LDAP_LIBS) $(ZLIB_LIBS) \
  		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/swat@EXEEXT@: $(BINARY_PREREQS) $(SWAT_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(SWAT_OBJ) $(DYNEXP) $(LIBS) \
! 		$(KRB5LIBS) $(LDAP_LIBS) $(ZLIB_LIBS) \
! 		$(AUTH_LIBS) $(PASSDB_LIBS) $(PRINT_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/rpcclient@EXEEXT@: $(BINARY_PREREQS) $(RPCCLIENT_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(RPCCLIENT_OBJ) $(DYNEXP) $(LIBS) \
! 		$(KRB5LIBS) $(LDAP_LIBS) $(ZLIB_LIBS) \
! 		$(PASSDB_LIBS) \
! 		$(TERMLDFLAGS) $(TERMLIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/smbclient@EXEEXT@: $(BINARY_PREREQS) $(CLIENT_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(CLIENT_OBJ) $(DYNEXP) $(LIBS) \
! 		$(KRB5LIBS) $(LDAP_LIBS) $(ZLIB_LIBS) \
! 		$(NSCD_LIBS) $(DNSSD_LIBS) \
! 		$(TERMLDFLAGS) $(TERMLIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/net@EXEEXT@: $(BINARY_PREREQS) $(NET_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT) $(LIBNETAPI)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(NET_OBJ) $(DYNEXP) $(LIBS) \
! 		$(KRB5LIBS) $(LDAP_LIBS) $(ZLIB_LIBS) \
! 		$(NSCD_LIBS) $(PASSDB_LIBS) $(UUID_LIBS) \
! 		$(TERMLDFLAGS) $(TERMLIBS) \
! 		$(LIBNETAPI_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/profiles@EXEEXT@: $(BINARY_PREREQS) $(PROFILES_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(PROFILES_OBJ) $(DYNEXP) $(LIBS) \
! 		$(LDAP_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/smbspool@EXEEXT@: $(BINARY_PREREQS) $(CUPS_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(CUPS_OBJ) $(DYNEXP) $(LIBS) \
! 		$(KRB5LIBS) $(LDAP_LIBS) $(ZLIB_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/testparm@EXEEXT@: $(BINARY_PREREQS) $(TESTPARM_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(TESTPARM_OBJ) $(DYNEXP) $(LIBS) \
! 		$(LDAP_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/smbta-util@EXEEXT@: $(BINARY_PREREQS) $(SMBTA_UTIL_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(SMBTA_UTIL_OBJ) $(DYNEXP) $(LIBS) \
! 		$(LDAP_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/smbstatus@EXEEXT@: $(BINARY_PREREQS) $(STATUS_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(STATUS_OBJ) $(DYNEXP) $(LIBS) \
! 		$(LDAP_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/smbcontrol@EXEEXT@: $(BINARY_PREREQS) $(SMBCONTROL_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -DUSING_SMBCONTROL -o $@ $(LDFLAGS) $(SMBCONTROL_OBJ) $(DYNEXP) $(LIBS) \
! 		$(LDAP_LIBS) \
! 		@LIBUNWIND_PTRACE@ \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/smbtree@EXEEXT@: $(BINARY_PREREQS) $(SMBTREE_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(SMBTREE_OBJ) $(DYNEXP) $(LIBS) \
! 		$(KRB5LIBS) $(LDAP_LIBS) $(ZLIB_LIBS) \
! 		$(NSCD_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/smbpasswd@EXEEXT@: $(BINARY_PREREQS) $(SMBPASSWD_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(SMBPASSWD_OBJ) $(DYNEXP) $(LIBS) \
! 		$(KRB5LIBS) $(LDAP_LIBS) $(ZLIB_LIBS) \
! 		$(PASSDB_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/pdbedit@EXEEXT@: $(BINARY_PREREQS) $(PDBEDIT_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(PDBEDIT_OBJ) $(DYNEXP) $(LIBS) \
! 		$(KRB5LIBS) $(LDAP_LIBS) $(ZLIB_LIBS) \
! 		$(PASSDB_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/smbget@EXEEXT@: $(BINARY_PREREQS) $(SMBGET_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(SMBGET_OBJ) $(DYNEXP) $(LIBS) \
! 		$(KRB5LIBS) $(LDAP_LIBS) $(ZLIB_LIBS) \
! 		$(NSCD_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/nmblookup@EXEEXT@: $(BINARY_PREREQS) $(NMBLOOKUP_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(NMBLOOKUP_OBJ) $(DYNEXP) $(LIBS) \
! 		$(KRB5LIBS) $(LDAP_LIBS) $(ZLIB_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/smbtorture@EXEEXT@: $(BINARY_PREREQS) $(SMBTORTURE_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(SMBTORTURE_OBJ) $(DYNEXP) $(LIBS) \
! 		$(KRB5LIBS) $(LDAP_LIBS) $(ZLIB_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/talloctort@EXEEXT@: $(BINARY_PREREQS) $(TALLOCTORT_OBJ) $(LIBTALLOC) $(LIBTDB)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(TALLOCTORT_OBJ) $(DYNEXP) $(LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS)
  
! bin/replacetort@EXEEXT@: $(BINARY_PREREQS) $(REPLACETORT_OBJ) $(LIBTALLOC) $(LIBTDB)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(REPLACETORT_OBJ) $(DYNEXP) $(LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS)
  
! bin/smbconftort@EXEEXT@: $(BINARY_PREREQS) $(SMBCONFTORT_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(SMBCONFTORT_OBJ) $(DYNEXP) $(LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/pthreadpooltest@EXEEXT@: $(BINARY_PREREQS) $(PTHREADPOOLTEST_OBJ)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(PTHREADPOOLTEST_OBJ) -lpthread
  
! bin/masktest@EXEEXT@: $(BINARY_PREREQS) $(MASKTEST_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(MASKTEST_OBJ) $(DYNEXP) $(LIBS) \
! 		$(KRB5LIBS) $(LDAP_LIBS) $(ZLIB_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/msgtest@EXEEXT@: $(BINARY_PREREQS) $(MSGTEST_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(MSGTEST_OBJ) $(DYNEXP) $(LIBS) \
! 		$(KRB5LIBS) $(LDAP_LIBS) $(ZLIB_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/smbcacls@EXEEXT@: $(BINARY_PREREQS) $(SMBCACLS_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(SMBCACLS_OBJ) $(DYNEXP) $(LIBS) \
! 		$(KRB5LIBS) $(LDAP_LIBS) $(ZLIB_LIBS) \
! 		$(NSCD_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/smbcquotas@EXEEXT@: $(BINARY_PREREQS) $(SMBCQUOTAS_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(SMBCQUOTAS_OBJ) $(DYNEXP) $(LIBS) \
! 		$(KRB5LIBS) $(LDAP_LIBS) $(ZLIB_LIBS) \
! 		$(NSCD_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/eventlogadm@EXEEXT@: $(BINARY_PREREQS) $(EVTLOGADM_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(EVTLOGADM_OBJ) $(DYNEXP) $(LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/sharesec@EXEEXT@: $(BINARY_PREREQS) $(SHARESEC_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(SHARESEC_OBJ) $(DYNEXP) $(LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/locktest@EXEEXT@: $(BINARY_PREREQS) $(LOCKTEST_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(LOCKTEST_OBJ) $(DYNEXP) $(LIBS) \
! 		$(KRB5LIBS) $(LDAP_LIBS) $(ZLIB_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
  bin/nsstest@EXEEXT@: $(BINARY_PREREQS) $(NSSTEST_OBJ)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(NSSTEST_OBJ) $(DYNEXP) $(LIBS)
  
! bin/pdbtest@EXEEXT@: $(BINARY_PREREQS) $(PDBTEST_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(PDBTEST_OBJ) $(DYNEXP) $(LIBS) \
! 		$(KRB5LIBS) $(LDAP_LIBS) $(ZLIB_LIBS) \
! 		$(PASSDB_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/vfstest@EXEEXT@: $(BINARY_PREREQS) $(VFSTEST_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(VFSTEST_OBJ) $(DYNEXP) $(LIBS) \
! 		$(KRB5LIBS) $(LDAP_LIBS) $(ZLIB_LIBS) \
! 		$(ACL_LIBS) $(AUTH_LIBS) $(AVAHI_LIBS) $(DNSSD_LIBS) \
! 		$(NSCD_LIBS) $(PRINT_LIBS) \
! 		$(TERMLDFLAGS) $(TERMLIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/smbiconv@EXEEXT@: $(BINARY_PREREQS) $(SMBICONV_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(SMBICONV_OBJ) $(DYNEXP) $(LIBS) \
! 		$(TERMLDFLAGS) $(TERMLIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/log2pcap@EXEEXT@: $(BINARY_PREREQS) $(LOG2PCAP_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(LOG2PCAP_OBJ) $(DYNEXP) $(LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/locktest2@EXEEXT@: $(BINARY_PREREQS) $(LOCKTEST2_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(LOCKTEST2_OBJ) $(DYNEXP) $(LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/debug2html@EXEEXT@: $(BINARY_PREREQS) $(DEBUG2HTML_OBJ) $(LIBTALLOC)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(DEBUG2HTML_OBJ) $(DYNEXP) $(LIBS) \
! 		$(LIBTALLOC_LIBS)
  
! bin/smbfilter@EXEEXT@: $(BINARY_PREREQS) $(SMBFILTER_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(SMBFILTER_OBJ) $(DYNEXP) $(LIBS) \
! 		$(KRB5LIBS) $(LDAP_LIBS) $(ZLIB_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
  bin/versiontest: $(BINARY_PREREQS) lib/version_test.o $(VERSION_OBJ)
  	@echo Linking $@
***************
*** 2204,2221 ****
  #-------------------------------------------------------------------
  
  LIBWBCLIENT_OBJ0 = ../nsswitch/libwbclient/wbclient.o \
! 		  ../nsswitch/libwbclient/wbc_util.o \
! 		  ../nsswitch/libwbclient/wbc_pwd.o \
! 		  ../nsswitch/libwbclient/wbc_idmap.o \
! 		  ../nsswitch/libwbclient/wbc_sid.o \
! 		  ../nsswitch/libwbclient/wbc_guid.o \
! 		  ../nsswitch/libwbclient/wbc_pam.o
  
  LIBWBCLIENT_OBJ = $(LIBWBCLIENT_OBJ0) \
- 		  $(WBCOMMON_OBJ) \
- 		  $(LIBREPLACE_OBJ) \
  		  ../lib/util/blocking.o \
! 		  $(SOCKET_WRAPPER_OBJ)
  
  LIBWBCLIENT_SHARED_TARGET=@LIBWBCLIENT_SHARED_TARGET@
  LIBWBCLIENT_SOVER=@LIBWBCLIENT_SOVER@
--- 2062,2078 ----
  #-------------------------------------------------------------------
  
  LIBWBCLIENT_OBJ0 = ../nsswitch/libwbclient/wbclient.o \
! 		   ../nsswitch/libwbclient/wbc_guid.o \
! 		   ../nsswitch/libwbclient/wbc_idmap.o \
! 		   ../nsswitch/libwbclient/wbc_pam.o \
! 		   ../nsswitch/libwbclient/wbc_pwd.o \
! 		   ../nsswitch/libwbclient/wbc_sid.o \
! 		   ../nsswitch/libwbclient/wbc_util.o
  
  LIBWBCLIENT_OBJ = $(LIBWBCLIENT_OBJ0) \
  		  ../lib/util/blocking.o \
! 		  $(WBCOMMON_OBJ) \
! 		  $(LIBREPLACE_OBJ) $(SOCKET_WRAPPER_OBJ)
  
  LIBWBCLIENT_SHARED_TARGET=@LIBWBCLIENT_SHARED_TARGET@
  LIBWBCLIENT_SOVER=@LIBWBCLIENT_SOVER@
***************
*** 2223,2229 ****
  LIBWBCLIENT_SHARED_TARGET_SONAME=@LIBWBCLIENT_SHARED_TARGET_SOVER@
  LIBWBCLIENT_STATIC_TARGET=@LIBWBCLIENT_STATIC_TARGET@
  LIBWBCLIENT_SYMS=$(srcdir)/exports/libwbclient.@SYMSEXT@
! LIBWBCLIENT_HEADERS=$(srcdir)/../nsswitch/libwbclient/wbclient.h
  
  $(LIBWBCLIENT_SYMS): $(LIBWBCLIENT_HEADERS)
  	@$(MKSYMS_SH) $(AWK) $@ $(LIBWBCLIENT_HEADERS)
--- 2080,2087 ----
  LIBWBCLIENT_SHARED_TARGET_SONAME=@LIBWBCLIENT_SHARED_TARGET_SOVER@
  LIBWBCLIENT_STATIC_TARGET=@LIBWBCLIENT_STATIC_TARGET@
  LIBWBCLIENT_SYMS=$(srcdir)/exports/libwbclient.@SYMSEXT@
! LIBWBCLIENT_HEADERS=$(srcdir)/../nsswitch/libwbclient/wbclient.h \
! 		    $(srcdir)/../nsswitch/winbind_client.h
  
  $(LIBWBCLIENT_SYMS): $(LIBWBCLIENT_HEADERS)
  	@$(MKSYMS_SH) $(AWK) $@ $(LIBWBCLIENT_HEADERS)
***************
*** 2370,2410 ****
  #
  #-------------------------------------------------------------------
  
! LIBNETAPI_OBJ0 = lib/netapi/netapi.o \
! 		 lib/netapi/cm.o \
! 		 librpc/gen_ndr/ndr_libnetapi.o \
! 		 lib/netapi/libnetapi.o \
! 		 lib/netapi/joindomain.o \
! 		 lib/netapi/serverinfo.o \
! 		 lib/netapi/getdc.o \
! 		 lib/netapi/user.o \
! 		 lib/netapi/group.o \
! 		 lib/netapi/localgroup.o \
! 		 lib/netapi/samr.o \
! 		 lib/netapi/sid.o \
! 		 lib/netapi/share.o \
! 		 lib/netapi/file.o \
! 		 lib/netapi/shutdown.o \
! 		 lib/netapi/netlogon.o
! 
! LIBNETAPI_OBJ  = $(LIBNETAPI_OBJ0) $(LIBNET_OBJ) \
! 		 $(LIBSMBCONF_OBJ) \
! 		 $(REG_SMBCONF_OBJ) \
! 		 $(PARAM_WITHOUT_REG_OBJ) $(LIB_NONSMBD_OBJ) \
! 		 $(LIBSMB_OBJ) $(KRBCLIENT_OBJ) \
! 		 $(LIBMSRPC_OBJ) $(LIBMSRPC_GEN_OBJ) \
! 		 $(PASSDB_OBJ) $(SMBLDAP_OBJ) $(GROUPDB_OBJ) \
! 		 $(DCUTIL_OBJ) $(LIBADS_OBJ) $(PRIVILEGES_BASIC_OBJ) \
! 		 auth/token_util.o \
! 		 $(LIBCLI_INITSHUTDOWN_OBJ) \
! 		 $(LIBCLI_WKSSVC_OBJ) \
! 		 $(LIBCLI_SRVSVC_OBJ) \
! 		 $(LIBCLI_LSA_OBJ) \
! 		 $(LIBCLI_SAMR_OBJ) \
! 		 $(LIBCLI_NETLOGON_OBJ) \
! 		 $(RPC_CLIENT_SCHANNEL_OBJ) \
! 		 rpc_client/init_netlogon.o \
! 		 rpc_client/init_samr.o
  
  LIBNETAPI_SHARED_TARGET=@LIBNETAPI_SHARED_TARGET@
  LIBNETAPI_SOVER=@LIBNETAPI_SOVER@
--- 2228,2249 ----
  #
  #-------------------------------------------------------------------
  
! LIBNETAPI_OBJ = lib/netapi/netapi.o \
! 		lib/netapi/cm.o \
! 		librpc/gen_ndr/ndr_libnetapi.o \
! 		lib/netapi/libnetapi.o \
! 		lib/netapi/joindomain.o \
! 		lib/netapi/serverinfo.o \
! 		lib/netapi/getdc.o \
! 		lib/netapi/user.o \
! 		lib/netapi/group.o \
! 		lib/netapi/localgroup.o \
! 		lib/netapi/samr.o \
! 		lib/netapi/sid.o \
! 		lib/netapi/share.o \
! 		lib/netapi/file.o \
! 		lib/netapi/shutdown.o \
! 		lib/netapi/netlogon.o
  
  LIBNETAPI_SHARED_TARGET=@LIBNETAPI_SHARED_TARGET@
  LIBNETAPI_SOVER=@LIBNETAPI_SOVER@
***************
*** 2417,2436 ****
  $(LIBNETAPI_SYMS): $(LIBNETAPI_HEADERS)
  	@$(MKSYMS_SH) $(AWK) $@ $(LIBNETAPI_HEADERS)
  
! $(LIBNETAPI_SHARED_TARGET_SONAME): $(BINARY_PREREQS) $(LIBNETAPI_OBJ) $(LIBNETAPI_SYMS) $(LIBTALLOC) $(LIBTDB) $(LIBWBCLIENT)
  	@echo Linking shared library $@
  	@$(SHLD_DSO) $(LIBNETAPI_OBJ) \
- 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(LIBWBCLIENT_LIBS) $(LIBS) \
  		$(LDAP_LIBS) $(KRB5LIBS) $(NSCD_LIBS) $(ZLIB_LIBS) \
  		@SONAMEFLAG@`basename $@`
  
  $(LIBNETAPI_SHARED_TARGET): $(LIBNETAPI_SHARED_TARGET_SONAME)
  	@rm -f $@
  	@ln -f -s `basename $(LIBNETAPI_SHARED_TARGET_SONAME)` $@
  
! $(LIBNETAPI_STATIC_TARGET): $(BINARY_PREREQS) $(LIBNETAPI_OBJ0)
  	@echo Linking non-shared library $@
! 	@-$(AR) -rc $@ $(LIBNETAPI_OBJ0)
  
  libnetapi: $(LIBNETAPI)
  
--- 2256,2276 ----
  $(LIBNETAPI_SYMS): $(LIBNETAPI_HEADERS)
  	@$(MKSYMS_SH) $(AWK) $@ $(LIBNETAPI_HEADERS)
  
! $(LIBNETAPI_SHARED_TARGET_SONAME): $(BINARY_PREREQS) $(LIBNETAPI_OBJ) $(LIBNETAPI_SYMS) $(LIBTALLOC) $(LIBTDB) $(LIBWBCLIENT) $(LIBSMBCLIENT)
  	@echo Linking shared library $@
  	@$(SHLD_DSO) $(LIBNETAPI_OBJ) \
  		$(LDAP_LIBS) $(KRB5LIBS) $(NSCD_LIBS) $(ZLIB_LIBS) \
+ 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(LIBS) \
+ 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  		@SONAMEFLAG@`basename $@`
  
  $(LIBNETAPI_SHARED_TARGET): $(LIBNETAPI_SHARED_TARGET_SONAME)
  	@rm -f $@
  	@ln -f -s `basename $(LIBNETAPI_SHARED_TARGET_SONAME)` $@
  
! $(LIBNETAPI_STATIC_TARGET): $(BINARY_PREREQS) $(LIBNETAPI_OBJ)
  	@echo Linking non-shared library $@
! 	@-$(AR) -rc $@ $(LIBNETAPI_OBJ)
  
  libnetapi: $(LIBNETAPI)
  
***************
*** 2495,2502 ****
  LIBSMBCLIENT_SYMS=$(srcdir)/exports/libsmbclient.@SYMSEXT@
  LIBSMBCLIENT_HEADERS=$(srcdir)/include/libsmbclient.h
  
  $(LIBSMBCLIENT_SYMS): $(LIBSMBCLIENT_HEADERS)
! 	@$(MKSYMS_SH) $(AWK) $@ $(LIBSMBCLIENT_HEADERS)
  
  $(LIBSMBCLIENT_SHARED_TARGET_SONAME): $(BINARY_PREREQS) $(LIBSMBCLIENT_OBJ) $(LIBSMBCLIENT_THREAD_OBJ) $(LIBSMBCLIENT_SYMS) $(LIBTALLOC) $(LIBTDB) $(LIBWBCLIENT)
  	@echo Linking shared library $@
--- 2335,2344 ----
  LIBSMBCLIENT_SYMS=$(srcdir)/exports/libsmbclient.@SYMSEXT@
  LIBSMBCLIENT_HEADERS=$(srcdir)/include/libsmbclient.h
  
+ # export all symbols not that this is a common lib for all programs
  $(LIBSMBCLIENT_SYMS): $(LIBSMBCLIENT_HEADERS)
! 	#@$(MKSYMS_SH) $(AWK) $@ $(LIBSMBCLIENT_HEADERS)
! 	@echo "{};" > $@
  
  $(LIBSMBCLIENT_SHARED_TARGET_SONAME): $(BINARY_PREREQS) $(LIBSMBCLIENT_OBJ) $(LIBSMBCLIENT_THREAD_OBJ) $(LIBSMBCLIENT_SYMS) $(LIBTALLOC) $(LIBTDB) $(LIBWBCLIENT)
  	@echo Linking shared library $@
***************
*** 2509,2517 ****
  	@rm -f $@
  	@ln -s `basename $(LIBSMBCLIENT_SHARED_TARGET_SONAME)` $@
  
! $(LIBSMBCLIENT_STATIC_TARGET): $(BINARY_PREREQS) $(LIBSMBCLIENT_OBJ1)
  	@echo Linking non-shared library $@
! 	@-$(AR) -rc $@ $(LIBSMBCLIENT_OBJ1)
  
  libsmbclient: $(LIBSMBCLIENT)
  
--- 2351,2359 ----
  	@rm -f $@
  	@ln -s `basename $(LIBSMBCLIENT_SHARED_TARGET_SONAME)` $@
  
! $(LIBSMBCLIENT_STATIC_TARGET): $(BINARY_PREREQS) $(LIBSMBCLIENT_OBJ)
  	@echo Linking non-shared library $@
! 	@-$(AR) -rc $@ $(LIBSMBCLIENT_OBJ)
  
  libsmbclient: $(LIBSMBCLIENT)
  
***************
*** 2648,2673 ****
  
  showlibs:: showlibsmbsharemodes
  
- #-------------------------------------------------------------------
- #
- # libbigballofmud
- #
- #-------------------------------------------------------------------
- 
- # This is probably wrong for anything other than the GNU linker.
- bin/libbigballofmud.@SHLIBEXT@: $(BINARY_PREREQS) $(LIBBIGBALLOFMUD_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBWBCLIENT)
- 	@echo Linking shared library $@
- 	@$(SHLD) $(LDSHFLAGS) -o $@ $(LIBBIGBALLOFMUD_OBJ) \
- 		$(LIBS) $(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(LIBWBCLIENT_LIBS) \
- 		$(PASSDB_LIBS) $(IDMAP_LIBS) $(KRB5LIBS) $(LDAP_LIBS) -lz \
- 		@SONAMEFLAG@`basename $@`.$(LIBBIGBALLOFMUD_MAJOR)
- 	ln -snf libbigballofmud.so bin/libbigballofmud.so.0
- 
- # It would be nice to build a static bigballofmud too, but when I try
- # I get linker errors about dl_open and similar things.  I'm not sure if
- # it can be fixed or if they just can't be called from a static
- # library.
- 
  #####################################################################
  #
  # shared modules
--- 2490,2495 ----
***************
*** 2737,2752 ****
  	@echo "Linking $@"
  	@$(SHLD_LIBRARY) $(RPC_RPCECHO_OBJ)
  
! bin/winbindd@EXEEXT@: $(BINARY_PREREQS) $(WINBINDD_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB) $(LIBWBCLIENT)
! 	@echo "Linking $@"
! 	@$(CC) -o $@ $(WINBINDD_OBJ) $(LDFLAGS) $(DYNEXP) $(LIBS) \
! 		$(POPT_LIBS) $(KRB5LIBS) $(LDAP_LIBS) $(AUTH_LIBS) \
! 		$(PASSDB_LIBS) $(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(LIBWBCLIENT_LIBS) \
! 		$(ZLIB_LIBS)
  
  bin/vlp@EXEEXT@: $(BINARY_PREREQS) $(VLP_OBJ) $(LIBTDB)
  	@echo "Linking $@"
! 	@$(CC) -o $@ $(VLP_OBJ) $(LDFLAGS) $(DYNEXP) \
  		$(TERMLDFLAGS) $(TERMLIBS) $(LIBS) $(POPT_LIBS) \
  		$(LIBTDB_LIBS)
  
--- 2559,2575 ----
  	@echo "Linking $@"
  	@$(SHLD_LIBRARY) $(RPC_RPCECHO_OBJ)
  
! bin/winbindd@EXEEXT@: $(BINARY_PREREQS) $(WINBINDD_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
! 	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(WINBINDD_OBJ) $(DYNEXP) $(LIBS) \
! 		$(KRB5LIBS) $(LDAP_LIBS) $(ZLIB_LIBS) \
! 		$(AUTH_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
  bin/vlp@EXEEXT@: $(BINARY_PREREQS) $(VLP_OBJ) $(LIBTDB)
  	@echo "Linking $@"
! 	@$(CC) -o $@ $(LDFLAGS) $(VLP_OBJ) $(DYNEXP) \
  		$(TERMLDFLAGS) $(TERMLIBS) $(LIBS) $(POPT_LIBS) \
  		$(LIBTDB_LIBS)
  
***************
*** 2756,2765 ****
  		$(LIBREPLACE_LIBS) $(PTHREAD_LDFLAGS) \
  		@SONAMEFLAG@`basename $@`@NSSSONAMEVERSIONSUFFIX@
  
! @WINBIND_WINS_NSS@: $(BINARY_PREREQS) $(WINBIND_WINS_NSS_OBJ) $(LIBTALLOC) $(LIBTDB)
  	@echo "Linking $@"
! 	@$(SHLD) $(LDSHFLAGS) -o $@ $(WINBIND_WINS_NSS_OBJ) \
! 		$(LDAP_LIBS) $(KRB5LIBS) $(LIBS) $(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(ZLIB_LIBS) \
  		@SONAMEFLAG@`basename $@`@NSSSONAMEVERSIONSUFFIX@
  
  bin/winbind_krb5_locator.@SHLIBEXT@: $(BINARY_PREREQS) $(WINBIND_KRB5_LOCATOR_OBJ) $(LIBWBCLIENT)
--- 2579,2590 ----
  		$(LIBREPLACE_LIBS) $(PTHREAD_LDFLAGS) \
  		@SONAMEFLAG@`basename $@`@NSSSONAMEVERSIONSUFFIX@
  
! @WINBIND_WINS_NSS@: $(BINARY_PREREQS) $(WINBIND_WINS_NSS_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT)
  	@echo "Linking $@"
! 	@$(SHLD) $(LDSHFLAGS) -o $@ $(WINBIND_WINS_NSS_OBJ) $(LIBS) \
! 		$(KRB5LIBS) $(LDAP_LIBS) $(ZLIB_LIBS) \
!                 $(LIBTALLOC_LIBS) $(LIBTDB_LIBS) \
!                 $(LIBSMBCLIENT_LIBS) \
  		@SONAMEFLAG@`basename $@`@NSSSONAMEVERSIONSUFFIX@
  
  bin/winbind_krb5_locator.@SHLIBEXT@: $(BINARY_PREREQS) $(WINBIND_KRB5_LOCATOR_OBJ) $(LIBWBCLIENT)
***************
*** 3099,3208 ****
  ## None here right now
  #########################################################
  
! bin/wbinfo@EXEEXT@: $(BINARY_PREREQS) $(WBINFO_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB) $(LIBWBCLIENT)
  	@echo Linking $@
  	@$(CC) -o $@ $(LDFLAGS) $(WBINFO_OBJ) $(DYNEXP) $(LIBS) \
! 		$(LDAP_LIBS) $(POPT_LIBS) $(LIBTALLOC_LIBS) $(LIBTDB_LIBS) \
! 		$(LIBWBCLIENT_LIBS)
  
! bin/ntlm_auth@EXEEXT@: $(BINARY_PREREQS) $(NTLM_AUTH_OBJ) $(PARAM_OBJ) \
! 	$(LIB_NONSMBD_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(DYNEXP) $(NTLM_AUTH_OBJ) \
! 		$(PARAM_OBJ) $(LIB_NONSMBD_OBJ) $(LIBS) \
! 		$(POPT_LIBS) $(KRB5LIBS) $(LDAP_LIBS) $(NSCD_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(LIBWBCLIENT_LIBS) @INIPARSERLIBS@
  
! bin/pam_smbpass.@SHLIBEXT@: $(BINARY_PREREQS) $(PAM_SMBPASS_OBJ) $(LIBCLI_LDAP_NDR_OBJ) $(LIBTALLOC) $(LIBWBCLIENT) $(LIBTDB)
  	@echo "Linking shared library $@"
  	@$(SHLD) $(LDSHFLAGS) -o $@ $(PAM_SMBPASS_OBJ) $(LIBCLI_LDAP_NDR_OBJ) -lpam $(DYNEXP) \
  		$(LIBS) $(LDAP_LIBS) $(NSCD_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(LIBWBCLIENT_LIBS)
  
  bin/tdbbackup@EXEEXT@: $(BINARY_PREREQS) $(TDBBACKUP_OBJ) $(LIBTALLOC) $(LIBTDB)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(TDBBACKUP_OBJ) $(DYNEXP) \
! 		$(LIBS) $(LIBTALLOC_LIBS) $(LIBTDB_LIBS)
  
  bin/tdbtool@EXEEXT@: $(BINARY_PREREQS) $(TDBTOOL_OBJ) $(LIBTALLOC) $(LIBTDB)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(TDBTOOL_OBJ) $(DYNEXP) \
! 		$(LIBS) $(LIBTALLOC_LIBS) $(LIBTDB_LIBS)
  
  bin/tdbdump@EXEEXT@: $(BINARY_PREREQS) $(TDBDUMP_OBJ) $(LIBTDB)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(TDBDUMP_OBJ) $(DYNEXP) $(LIBTDB_LIBS) $(LIBREPLACE_LIBS)
  
  bin/tdbrestore@EXEEXT@: $(BINARY_PREREQS) $(TDBRESTORE_OBJ) $(LIBTDB)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(TDBRESTORE_OBJ) $(DYNEXP) $(LIBTDB_LIBS) $(LIBREPLACE_LIBS)
  
  bin/tdbtorture@EXEEXT@: $(BINARY_PREREQS) $(TDBTORTURE_OBJ) $(LIBTALLOC) $(LIBTDB)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(TDBTORTURE_OBJ) $(DYNEXP) \
! 		$(LIBS) $(LIBTALLOC_LIBS) $(LIBTDB_LIBS)
  
! bin/t_strcmp@EXEEXT@: $(BINARY_PREREQS) $(LIBTALLOC) bin/libbigballofmud.@SHLIBEXT@ torture/t_strcmp.o
! 	$(CC) -o $@ $(DYNEXP) $(LIBS) $(LIBTALLOC_LIBS) \
! 		torture/t_strcmp.o -L ./bin -lbigballofmud $(LIBWBCLIENT_LIBS)
! 
! bin/t_strstr@EXEEXT@: $(BINARY_PREREQS) $(LIBTALLOC) bin/libbigballofmud.@SHLIBEXT@ torture/t_strstr.o
! 	$(CC) -o $@ $(DYNEXP) $(LIBS) $(LIBTALLOC_LIBS) \
! 		torture/t_strstr.o -L ./bin -lbigballofmud $(LIBWBCLIENT_LIBS)
! 
! bin/t_strappend@EXEEXT@: $(BINARY_PREREQS) $(LIBTALLOC) bin/libbigballofmud.@SHLIBEXT@ torture/t_strappend.o
! 	$(CC) -o $@ $(DYNEXP) $(LIBS) $(LIBTALLOC_LIBS) \
! 		torture/t_strappend.o -L ./bin -lbigballofmud $(LIBWBCLIENT_LIBS)
! 
! bin/t_stringoverflow@EXEEXT@: $(BINARY_PREREQS) bin/libbigballofmud.@SHLIBEXT@ torture/t_stringoverflow.o
! 	$(CC) -o $@ $(DYNEXP) torture/t_stringoverflow.o \
! 		-L./bin -lbigballofmud $(LIBWBCLIENT_LIBS)
  
  bin/timelimit@EXEEXT@: script/tests/timelimit.o
  	@echo Linking $@
  	@$(CC) -o $@ $(DYNEXP) script/tests/timelimit.o
  
! bin/rpc_open_tcp@EXEEXT@: $(BINARY_PREREQS) $(RPC_OPEN_TCP_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBWBCLIENT)
  	@echo "Linking $@"
! 	@$(CC) -o $@ $(RPC_OPEN_TCP_OBJ) $(LDFLAGS) $(DYNEXP) \
! 		$(LIBS) $(LIBTALLOC_LIBS) $(LIBWBCLIENT_LIBS) \
! 		$(LDAP_LIBS) $(KRB5LIBS) $(NSCD_LIBS)
  
! bin/test_lp_load@EXEEXT@: $(BINARY_PREREQS) $(TEST_LP_LOAD_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB) $(LIBWBCLIENT)
  	@echo "Linking $@"
! 	@$(CC) -o $@ $(TEST_LP_LOAD_OBJ) \
! 		$(LDFLAGS) $(DYNEXP) $(LIBS) \
! 		$(LDAP_LIBS) $(LIBWBCLIENT_LIBS) \
! 		$(POPT_LIBS) $(LIBTALLOC_LIBS) $(LIBTDB_LIBS)
  
! bin/dbwrap_tool@EXEEXT@: $(DBWRAP_TOOL_OBJ) $(LIBTALLOC) $(LIBTDB)
  	@echo "Linking $@"
! 	@$(CC) -o $@ $(DBWRAP_TOOL_OBJ)\
! 		$(LDFLAGS) $(DYNEXP) $(LIBS) \
  		$(LDAP_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS)
  
  install-dbwrap_tool:: bin/dbwrap_tool@EXEEXT@
  	@$(SHELL) script/installbin.sh $(INSTALLPERMS_BIN) $(DESTDIR) $(prefix) $(BINDIR) $<
  
! bin/dbwrap_torture@EXEEXT@: $(DBWRAP_TORTURE_OBJ) $(LIBTALLOC) $(LIBTDB)
  	@echo "Linking $@"
! 	@$(CC) -o $@ $(DBWRAP_TORTURE_OBJ)\
! 		$(LDFLAGS) $(DYNEXP) $(LIBS) \
  		$(LDAP_LIBS) \
! 		$(POPT_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS)
  
  install-dbwrap_torture:: bin/dbwrap_torture@EXEEXT@
  	@$(SHELL) script/installbin.sh $(INSTALLPERMS_BIN) $(DESTDIR) $(prefix) $(BINDIR) $<
  
! bin/split_tokens@EXEEXT@: $(BINARY_PREREQS) $(SPLIT_TOKENS_OBJ) @BUILD_POPT@ $(LIBTALLOC) $(LIBTDB) $(LIBWBCLIENT)
  	@echo "Linking $@"
! 	@$(CC) -o $@ $(SPLIT_TOKENS_OBJ) \
! 		$(LDFLAGS) $(DYNEXP) \
! 		$(LIBS) $(POPT_LIBS) \
! 		$(LDAP_LIBS) $(LIBWBCLIENT_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS)
  
  install:: installservers installbin installman \
  		installscripts installdat installmodules @SWAT_INSTALL_TARGETS@ \
--- 2924,3039 ----
  ## None here right now
  #########################################################
  
! bin/wbinfo@EXEEXT@: $(BINARY_PREREQS) $(WBINFO_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo Linking $@
  	@$(CC) -o $@ $(LDFLAGS) $(WBINFO_OBJ) $(DYNEXP) $(LIBS) \
! 		$(LDAP_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/ntlm_auth@EXEEXT@: $(BINARY_PREREQS) $(NTLM_AUTH_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(NTLM_AUTH_OBJ) $(DYNEXP) $(LIBS) \
!                 $(KRB5LIBS) $(LDAP_LIBS) $(NSCD_LIBS) \
! 		@INIPARSERLIBS@ \
!                 $(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
!                 $(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/pam_smbpass.@SHLIBEXT@: $(BINARY_PREREQS) $(PAM_SMBPASS_OBJ) $(LIBCLI_LDAP_NDR_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo "Linking shared library $@"
  	@$(SHLD) $(LDSHFLAGS) -o $@ $(PAM_SMBPASS_OBJ) $(LIBCLI_LDAP_NDR_OBJ) -lpam $(DYNEXP) \
  		$(LIBS) $(LDAP_LIBS) $(NSCD_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
  bin/tdbbackup@EXEEXT@: $(BINARY_PREREQS) $(TDBBACKUP_OBJ) $(LIBTALLOC) $(LIBTDB)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(TDBBACKUP_OBJ) $(DYNEXP) $(LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS)
  
  bin/tdbtool@EXEEXT@: $(BINARY_PREREQS) $(TDBTOOL_OBJ) $(LIBTALLOC) $(LIBTDB)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(TDBTOOL_OBJ) $(DYNEXP) $(LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS)
  
  bin/tdbdump@EXEEXT@: $(BINARY_PREREQS) $(TDBDUMP_OBJ) $(LIBTDB)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(TDBDUMP_OBJ) $(DYNEXP) $(LIBS) \
! 		$(LIBTDB_LIBS) $(LIBREPLACE_LIBS)
  
  bin/tdbrestore@EXEEXT@: $(BINARY_PREREQS) $(TDBRESTORE_OBJ) $(LIBTDB)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(TDBRESTORE_OBJ) $(DYNEXP) $(LIBS) \
! 		$(LIBTDB_LIBS) $(LIBREPLACE_LIBS)
  
  bin/tdbtorture@EXEEXT@: $(BINARY_PREREQS) $(TDBTORTURE_OBJ) $(LIBTALLOC) $(LIBTDB)
  	@echo Linking $@
! 	@$(CC) -o $@ $(LDFLAGS) $(TDBTORTURE_OBJ) $(DYNEXP) $(LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS)
! 
! bin/t_strcmp@EXEEXT@: $(BINARY_PREREQS) torture/t_strcmp.o $(LIBTALLOC) $(LIBSMBCLIENT)
! 	$(CC) -o $@ $(LDFLAGS) torture/t_strcmp.o $(DYNEXP) $(LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS)
  
! bin/t_strstr@EXEEXT@: $(BINARY_PREREQS) torture/t_strstr.o $(LIBTALLOC) $(LIBSMBCLIENT)
! 	$(CC) -o $@ $(LDFLAGS) torture/t_strstr.o $(DYNEXP) $(LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS)
! 
! bin/t_strappend@EXEEXT@: $(BINARY_PREREQS) torture/t_strappend.o $(LIBTALLOC) $(LIBSMBCLIENT)
! 	$(CC) -o $@ $(LDFLAGS) torture/t_strappend.o $(DYNEXP) $(LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS)
! 
! bin/t_stringoverflow@EXEEXT@: $(BINARY_PREREQS) torture/t_stringoverflow.o $(LIBTALLOC) $(LIBSMBCLIENT)
! 	$(CC) -o $@ $(LDFLAGS) torture/t_stringoverflow.o $(DYNEXP) $(LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS)
  
  bin/timelimit@EXEEXT@: script/tests/timelimit.o
  	@echo Linking $@
  	@$(CC) -o $@ $(DYNEXP) script/tests/timelimit.o
  
! bin/rpc_open_tcp@EXEEXT@: $(BINARY_PREREQS) $(RPC_OPEN_TCP_LOAD_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo "Linking $@"
! 	@$(CC) -o $@ $(LDFLAGS) $(RPC_OPEN_TCP_OBJ) $(DYNEXP) $(LIBS) \
! 		$(KRB5LIBS) $(LDAP_LIBS) $(ZLIB_LIBS) \
!                 $(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
!                 $(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/test_lp_load@EXEEXT@: $(BINARY_PREREQS) $(TESTLP_LOAD_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo "Linking $@"
! 	@$(CC) -o $@ $(LDFLAGS) $(TEST_LP_LOAD_OBJ) $(DYNEXP) $(LIBS) \
! 		$(LDAP_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
! bin/dbwrap_tool@EXEEXT@: $(BINARY_PREREQS) $(DBWRAP_TOOL_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo "Linking $@"
! 	@$(CC) -o $@ $(LDFLAGS) $(DBWRAP_TOOL_OBJ) $(DYNEXP) $(LIBS) \
  		$(LDAP_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
  install-dbwrap_tool:: bin/dbwrap_tool@EXEEXT@
  	@$(SHELL) script/installbin.sh $(INSTALLPERMS_BIN) $(DESTDIR) $(prefix) $(BINDIR) $<
  
! bin/dbwrap_torture@EXEEXT@: $(BINARY_PREREQS) $(DBWRAP_TORTURE_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo "Linking $@"
! 	@$(CC) -o $@ $(LDFLAGS) $(DBWRAP_TORTURE_OBJ) $(DYNEXP) $(LIBS) \
  		$(LDAP_LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
  install-dbwrap_torture:: bin/dbwrap_torture@EXEEXT@
  	@$(SHELL) script/installbin.sh $(INSTALLPERMS_BIN) $(DESTDIR) $(prefix) $(BINDIR) $<
  
! bin/split_tokens@EXEEXT@: $(BINARY_PREREQS) $(SPLIT_TOKENS_OBJ) $(LIBTALLOC) $(LIBTDB) $(LIBSMBCLIENT) $(LIBWBCLIENT)
  	@echo "Linking $@"
! 	@$(CC) -o $@ $(LDFLAGS) $(SPLIT_TOKENS_OBJ) $(DYNEXP) $(LIBS) \
! 		$(LIBTALLOC_LIBS) $(LIBTDB_LIBS) $(POPT_LIBS) \
! 		$(LIBSMBCLIENT_LIBS) $(LIBWBCLIENT_LIBS)
  
  install:: installservers installbin installman \
  		installscripts installdat installmodules @SWAT_INSTALL_TARGETS@ \
