--- a/lispconf/Makefile
+++ b/lispconf/Makefile
@@ -1,14 +1,13 @@
 CC=gcc
 GENGETOPT=gengetopt
-CFLAGS=-I../lisp_mod -g
+CFLAGS+=-I../lisp_mod -g
 DEPS = ../lisp_mod/lisp_ipc.h
 LIBS = -lm
-LDFLAGS = $(LIBS)
 OBJS = cmdline.o lispconf.o 
 PREFIX = /usr/local/sbin
 
 lispconf: $(OBJS)
-	gcc -o $@ $^ $(CFLAGS) $(LDFLAGS)
+	$(CC) -o $@ $^ $(CFLAGS) $(LDFLAGS) $(LIBS)
 
 cmdline.c: lispconf.ggo
 	$(GENGETOPT) -i $<
--- a/lispd/Makefile
+++ b/lispd/Makefile
@@ -13,7 +13,7 @@
 
 CC		= gcc
 GENGETOPT	= gengetopt
-CFLAGS		= -Wall -Wno-implicit-function-declaration -g -DDEBUG=3
+CFLAGS		+= -Wall -Wno-implicit-function-declaration -g -DDEBUG=3
 LIBS		= -lconfuse -lssl -lcrypto -lrt
 INC		= lispd.h
 MAKEFILE	= Makefile
@@ -26,7 +26,7 @@ EXE		= lispd
 PREFIX		= /usr/local/sbin
 
 $(EXE): $(OBJS) 
-	$(CC) -o $@ $^ $(CFLAGS) $(LIBS)
+	$(CC) -o $@ $^ $(CFLAGS) $(LDFLAGS) $(LIBS)
 
 #
 #	gengetops generates this...
