--- a/lispd/lispd.c
+++ b/lispd/lispd.c
@@ -160,16 +160,17 @@ int main(int argc, char **argv)
      *  Unload/load LISP kernel modules
      */
 
-    system("/sbin/modprobe -r lisp lisp_int");
+    system("/sbin/rmmod lisp");
+    system("/sbin/rmmod lisp_int");
 
-    if (system("/sbin/modprobe lisp")) {
+    if (system("/sbin/insmod lisp")) {
         syslog(LOG_DAEMON, "Loading the 'lisp' kernel module failed! Exiting...");
         exit(EXIT_FAILURE);
     }
     syslog(LOG_DAEMON, "Loaded the 'lisp' kernel module");
     sleep(1);
 
-    if (system("/sbin/modprobe lisp_int")) {
+    if (system("/sbin/insmod lisp_int")) {
         syslog(LOG_DAEMON, "Loading the 'lisp_int' kernel module failed! Exiting...");
         exit(EXIT_FAILURE);
     }
--- a/lispd/lispd_iface_mgmt.c
+++ b/lispd/lispd_iface_mgmt.c
@@ -1626,7 +1626,8 @@ void exit_cleanup(void) {
     close(netlink_fd);
 
     /* Remove lisp modules */
-    system("/sbin/modprobe -r lisp lisp_int");
+    system("/sbin/rmmod lisp");
+    system("/sbin/rmmod lisp_int");
 
     /* Remove source routing ip rule */
     delete_source_routing(ctrl_iface, &source_rloc, NULL);
