diff -ru a/setup.py b/setup.py
--- a/setup.py	2013-04-15 09:38:36.702548442 -0700
+++ b/setup.py	2013-04-15 09:42:22.896089964 -0700
@@ -462,7 +462,6 @@
                 'allmydata.mutable',
                 'allmydata.scripts',
                 'allmydata.storage',
-                'allmydata.test',
                 'allmydata.util',
                 'allmydata.web',
                 'allmydata.web.static',
