diff -ru a/setup.py b/setup.py
--- a/setup.py	2013-04-15 09:48:28.238581562 -0700
+++ b/setup.py	2013-04-15 09:48:46.878708699 -0700
@@ -465,8 +465,7 @@
                 'allmydata.util',
                 'allmydata.web',
                 'allmydata.web.static',
-                'allmydata.windows',
-                'buildtest'],
+                'allmydata.windows'],
       classifiers=trove_classifiers,
       test_suite="allmydata.test",
       install_requires=install_requires,
