diff -urN dibbler-0.8.1/ClntIfaceMgr/ClntIfaceMgr.cpp dibbler-0.8.1p1/ClntIfaceMgr/ClntIfaceMgr.cpp
--- dibbler-0.8.1/ClntIfaceMgr/ClntIfaceMgr.cpp	2011-10-31 22:00:54.000000000 +0100
+++ dibbler-0.8.1p1/ClntIfaceMgr/ClntIfaceMgr.cpp	2012-02-10 22:13:13.906767310 +0100
@@ -549,7 +549,7 @@
             int offset = prefixLen/8;
             if (prefixLen%8)
                 offset++;
-            buf[offset] = x->getID();
+            buf[offset] = (*i)->getID();
         } else {
             // users with too much time that play with virtual interfaces are out of luck
             Log(Error) << "Something is wrong. Detected more than 256 interface." << LogEnd;
